diff -urN a/modules/access/live555.cpp b/modules/access/live555.cpp
--- a/modules/access/live555.cpp	2021-06-30 23:38:57.623622272 -0300
+++ b/modules/access/live555.cpp	2021-06-30 23:42:11.735773366 -0300
@@ -59,6 +59,7 @@
 #include <liveMedia.hh>
 #include <liveMedia_version.hh>
 #include <Base64.hh>
+#include <NetAddress.hh>
 
 extern "C" {
 #include "../access/mms/asf.h"  /* Who said ugly ? */
@@ -729,6 +730,7 @@
     unsigned const thresh = 200000; /* RTP reorder threshold .2 second (default .1) */
     const char     *p_sess_lang = NULL;
     const char     *p_lang;
+    struct sockaddr_storage addr;
 
     b_rtsp_tcp    = var_CreateGetBool( p_demux, "rtsp-tcp" ) ||
                     var_GetBool( p_demux, "rtsp-http" );
@@ -852,7 +854,8 @@
             if( !p_sys->b_multicast )
             {
                 /* We need different rollover behaviour for multicast */
-                p_sys->b_multicast = IsMulticastAddress( sub->connectionEndpointAddress() );
+                sub->getConnectionEndpointAddress(addr);
+                p_sys->b_multicast = IsMulticastAddress( addr );
             }
 
             tk = (live_track_t*)malloc( sizeof( live_track_t ) );
