diff -Nuar thunderbird-60.0/testing/mozbase/mozsystemmonitor/mozsystemmonitor/resourcemonitor.py thunderbird-60.0-fixed/testing/mozbase/mozsystemmonitor/mozsystemmonitor/resourcemonitor.py
--- thunderbird-60.0/testing/mozbase/mozsystemmonitor/mozsystemmonitor/resourcemonitor.py	2018-07-31 21:21:02.000000000 +0300
+++ thunderbird-60.0-fixed/testing/mozbase/mozsystemmonitor/mozsystemmonitor/resourcemonitor.py	2018-10-30 14:21:42.632666466 +0200
@@ -65,6 +65,9 @@
 def get_disk_io_counters():
     try:
         io_counters = psutil.disk_io_counters()
+
+        if io_counters is None:
+            return PsutilStub().disk_io_counters()
     except RuntimeError:
         io_counters = PsutilStub().disk_io_counters()
 
