--- a/m2r.py.old	2022-08-11 15:18:19.930365371 +0200
+++ b/m2r.py	2022-08-11 15:19:05.680559926 +0200
@@ -10,8 +10,8 @@
 
 from docutils import statemachine, nodes, io, utils
 from docutils.parsers import rst
-from docutils.core import ErrorString
-from docutils.utils import SafeString, column_width
+from docutils.utils.error_reporting import ErrorString, SafeString
+from docutils.utils import column_width
 import mistune
 
 if sys.version_info < (3, ):
