diff --git a/src/FbTk/XftFontImp.cc b/src/FbTk/XftFontImp.cc
index fa34404..7f1fb62 100644
--- a/src/FbTk/XftFontImp.cc
+++ b/src/FbTk/XftFontImp.cc
@@ -78,7 +78,7 @@ bool XftFontImp::load(const std::string &name) {
     // overflow'). to prevent something like this we detect the maximium
     // number of glyphs by calculating the amount of 'WW' (pretending a 'wide'
     // glyph) fitting into 32k pixels 
-    unsigned int tw = textWidth("WW", 2);
+    unsigned int tw = std::max<unsigned int>(textWidth("WW", 2), 1);
     m_maxlength = 0x8000 / tw;
 
     return true;
