This fixes bug #1055

diff --git a/src/Window.cc b/src/Window.cc
index 0f91a9e..4410ffe 100644
--- a/src/Window.cc
+++ b/src/Window.cc
@@ -956,11 +956,14 @@ bool FluxboxWindow::setCurrentClient(WinClient &client, bool setinput) {
     if (!button)
         return false;
 
+    if (!client.acceptsFocus())
+        setinput = false; // don't try
+
     WinClient *old = m_client;
     m_client = &client;
 
     bool ret = setinput && focus();
-    if (setinput) {
+    if (setinput && old->acceptsFocus()) {
         m_client = old;
         return ret;
     }
