diff --git a/rc/filetype/sh.kak b/rc/filetype/sh.kak
index 38e26bb..e763922 100644
--- a/rc/filetype/sh.kak
+++ b/rc/filetype/sh.kak
@@ -2,6 +2,10 @@ hook global BufCreate .*\.(z|ba|c|k|mk)?sh(rc|_profile)? %{
     set-option buffer filetype sh
 }
 
+hook global BufCreate .*\.ebuild %{
+    set-option buffer filetype sh
+}
+
 hook global WinSetOption filetype=sh %{
     require-module sh
     set-option window static_words %opt{sh_static_words}
