--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -939,8 +939,6 @@
 MYSQL_CHECK_SSL()
 MYSQL_CHECK_SSL_DLLS()
 
-# Add system/bundled editline.
-MYSQL_CHECK_EDITLINE()
 # Add libevent
 MYSQL_CHECK_LIBEVENT()
 # Add lz4 library
@@ -1001,7 +999,9 @@
  ENABLE_TESTING()
 ENDIF()
 
-ADD_SUBDIRECTORY(include)
+IF(NOT WITHOUT_CLIENTLIBS)
+  ADD_SUBDIRECTORY(include)
+ENDIF(NOT WITHOUT_CLIENTLIBS)
 ADD_SUBDIRECTORY(dbug)
 ADD_SUBDIRECTORY(strings)
 ADD_SUBDIRECTORY(vio)
@@ -1035,7 +1035,6 @@
   ADD_SUBDIRECTORY(unittest/mytap/t)
 ENDIF()
 
-ADD_SUBDIRECTORY(client)
 ADD_SUBDIRECTORY(utilities)
 ADD_SUBDIRECTORY(share)
 ADD_SUBDIRECTORY(libservices)
@@ -1044,9 +1043,19 @@
   ADD_SUBDIRECTORY(man)
 ENDIF()
 
+IF(NOT WITHOUT_TOOLS)
+  # Add system/bundled editline.
+  MYSQL_CHECK_EDITLINE()
+  ADD_SUBDIRECTORY(client)
+ENDIF(NOT WITHOUT_TOOLS)
+
 IF(NOT WITHOUT_SERVER)
   ADD_SUBDIRECTORY(testclients)
   ADD_SUBDIRECTORY(sql)
+  IF(WITHOUT_TOOLS)
+    # Add system/bundled editline.
+    MYSQL_CHECK_EDITLINE()
+  ENDIF(WITHOUT_TOOLS)
 ENDIF()
 
 # scripts/mysql_config depends on client and server targets loaded above.
--- a/libmysql/CMakeLists.txt
+++ b/libmysql/CMakeLists.txt
@@ -234,12 +234,17 @@
   LIST(APPEND LIBS_TO_MERGE auth_win_client)
 ENDIF()
 
-# LDAP authentication SASL client plugin
-MESSAGE(STATUS "Creating LDAP authentication SASL client library.")
-ADD_SUBDIRECTORY(authentication_ldap)
-  
-# Merge several convenience libraries into one big mysqlclient
-MERGE_CONVENIENCE_LIBRARIES(mysqlclient ${LIBS_TO_MERGE} COMPONENT Development)
+IF(WITHOUT_CLIENTLIBS)
+  # Merge several convenience libraries into one big mysqlclient
+  MERGE_CONVENIENCE_LIBRARIES(mysqlclient ${LIBS_TO_MERGE} COMPONENT Development SKIP_INSTALL)
+ELSE(WITHOUT_CLIENTLIBS)
+  # LDAP authentication SASL client plugin
+  MESSAGE(STATUS "Creating LDAP authentication SASL client library.")
+  ADD_SUBDIRECTORY(authentication_ldap)
+
+  # Merge several convenience libraries into one big mysqlclient
+  MERGE_CONVENIENCE_LIBRARIES(mysqlclient ${LIBS_TO_MERGE} COMPONENT Development)
+ENDIF(WITHOUT_CLIENTLIBS)
 TARGET_LINK_LIBRARIES(mysqlclient PRIVATE ${LIBS_TO_LINK})
 
 # Visual Studio users need debug  static library for debug projects
@@ -278,9 +283,15 @@
 IF(NOT DISABLE_SHARED)
   # Merge several convenience libraries into one big mysqlclient
   # and link them together into shared library.
-  MERGE_LIBRARIES_SHARED(libmysql ${LIBS_TO_MERGE}
-    EXPORTS ${CLIENT_API_FUNCTIONS} ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
-    COMPONENT SharedLibraries)
+  IF(WITHOUT_CLIENTLIBS)
+    MERGE_LIBRARIES_SHARED(libmysql ${LIBS_TO_MERGE}
+      EXPORTS ${CLIENT_API_FUNCTIONS} ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
+      COMPONENT SharedLibraries SKIP_INSTALL)
+  ELSE(WITHOUT_CLIENTLIBS)
+    MERGE_LIBRARIES_SHARED(libmysql ${LIBS_TO_MERGE}
+      EXPORTS ${CLIENT_API_FUNCTIONS} ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
+      COMPONENT SharedLibraries)
+  ENDIF(WITHOUT_CLIENTLIBS)
   TARGET_LINK_LIBRARIES(libmysql PRIVATE ${LIBS_TO_LINK})
 
   IF(WIN32)
--- a/scripts/CMakeLists.txt
+++ b/scripts/CMakeLists.txt
@@ -458,18 +458,23 @@
 
   IF(WITH_SYSTEMD)
     SET(BIN_SCRIPTS
-      mysql_config
       mysqldumpslow
       )
   ELSE()
     SET(BIN_SCRIPTS
-      mysql_config
       mysqldumpslow
       mysqld_multi
       mysqld_safe
       )
   ENDIF()
 
+  IF(NOT WITHOUT_CLIENTLIBS)
+    SET(BIN_SCRIPTS
+      ${BIN_SCRIPTS}
+      mysql_config
+      )
+  ENDIF(NOT WITHOUT_CLIENTLIBS)
+
   SET(PKGCONFIG_FILE ${LIBMYSQL_OS_OUTPUT_NAME}.pc)
   STRING(REGEX REPLACE "/mysql$" "" PKGCONFIG_DIR "${INSTALL_LIBDIR}")
   SET(PKGCONFIG_DIR "${PKGCONFIG_DIR}/pkgconfig")
@@ -484,7 +489,7 @@
     ${CMAKE_CURRENT_BINARY_DIR}/${PKGCONFIG_FILE}
     ESCAPE_QUOTES @ONLY)
 
-  IF(INSTALL_PKGCONFIGDIR)
+  IF(INSTALL_PKGCONFIGDIR AND NOT WITHOUT_CLIENTLIBS)
     MESSAGE(STATUS "INSTALL ${PKGCONFIG_FILE} ${INSTALL_PKGCONFIGDIR}")
     INSTALL(FILES
       ${CMAKE_CURRENT_BINARY_DIR}/${PKGCONFIG_FILE}
--- a/support-files/CMakeLists.txt
+++ b/support-files/CMakeLists.txt
@@ -67,7 +67,9 @@
     INSTALL(FILES magic DESTINATION ${inst_location} COMPONENT SupportFiles)
   ENDIF()
 
-  INSTALL(FILES mysql.m4 DESTINATION ${INSTALL_SHAREDIR}/aclocal COMPONENT Development)
+  IF(NOT WITHOUT_CLIENTLIBS)
+    INSTALL(FILES mysql.m4 DESTINATION ${INSTALL_SHAREDIR}/aclocal COMPONENT Development)
+  ENDIF(NOT WITHOUT_CLIENTLIBS)
   
   SET(bindir ${prefix}/${INSTALL_BINDIR})
   SET(sbindir ${prefix}/${INSTALL_SBINDIR})
--- a/utilities/CMakeLists.txt
+++ b/utilities/CMakeLists.txt
@@ -92,11 +92,19 @@
   ADD_DEPENDENCIES(comp_err copy_openssl_dlls)
 ENDIF()
 
-MYSQL_ADD_EXECUTABLE(my_print_defaults my_print_defaults.cc)
+IF(NOT WITHOUT_CLIENTLIBS)
+  MYSQL_ADD_EXECUTABLE(my_print_defaults my_print_defaults.cc)
+ELSE(NOT WITHOUT_CLIENTLIBS)
+  ADD_EXECUTABLE(my_print_defaults my_print_defaults.cc)
+ENDIF(NOT WITHOUT_CLIENTLIBS)
 TARGET_LINK_LIBRARIES(my_print_defaults  mysys mysys_ssl)
 SET_TARGET_PROPERTIES(my_print_defaults PROPERTIES LINKER_LANGUAGE CXX)
 
-MYSQL_ADD_EXECUTABLE(perror perror.cc)
+IF(NOT WITHOUT_CLIENTLIBS)
+  MYSQL_ADD_EXECUTABLE(perror perror.cc)
+ELSE(NOT WITHOUT_CLIENTLIBS)
+  ADD_EXECUTABLE(perror perror.cc)
+ENDIF(NOT WITHOUT_CLIENTLIBS)
 ADD_DEPENDENCIES(perror GenError)
 TARGET_LINK_LIBRARIES(perror mysys mysys_ssl)
 SET_TARGET_PROPERTIES(perror PROPERTIES LINKER_LANGUAGE CXX)
