diff -urN psad-2.4.6.orig/init-scripts/psad-init.gentoo psad-2.4.6/init-scripts/psad-init.gentoo
--- psad-2.4.6.orig/init-scripts/psad-init.gentoo	2018-07-31 17:41:59.000000000 -0700
+++ psad-2.4.6/init-scripts/psad-init.gentoo	2022-03-04 16:30:38.381386906 -0800
@@ -28,6 +32,15 @@
 		--exec /usr/sbin/psad \
 		--pidfile /var/run/psad/psad.pid
 	eend $? "Failed to start ${SVCNAME}"
+
+	ebegin "Starting psadwatchd"
+	start-stop-daemon \
+	        --start \
+		--quiet \
+		--name psadwatchd \
+		--exec /usr/sbin/psadwatchd \
+	        --pidfile /var/run/psad/psadwatchd.pid
+	eend $? "Failed to start psadwatchd"
 }
 
 stop() {
