diff -Nuar kaffeine-2.0.6/CMakeLists.txt kaffeine-2.0.6-fixed/CMakeLists.txt
--- kaffeine-2.0.6/CMakeLists.txt	2017-02-21 17:00:38.000000000 +0000
+++ kaffeine-2.0.6-fixed/CMakeLists.txt	2017-07-30 13:46:32.287265804 +0000
@@ -93,7 +93,9 @@
 add_subdirectory(profiles)
 add_subdirectory(src)
 if (IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/po")
-  ki18n_install(po)
+	if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/po/CMakeLists.txt" )
+		add_subdirectory(po)
+	endif()
 endif()
 
 if(KF5DocTools_VERSION)
