--- a/man/CMakeLists.txt	2019-10-01 12:47:22.068412084 -0500
+++ b/man/CMakeLists.txt	2019-10-01 13:00:21.057628020 -0500
@@ -13,7 +13,12 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1335 USA
 
-SET(MAN1_SERVER innochecksum.1 my_print_defaults.1 myisam_ftdump.1 myisamchk.1
+IF(NOT WITHOUT_CLIENTLIBS)
+SET(MAN1_SERVER_CLIENTLIBS my_print_defaults.1 perror.1)
+ELSE(NOT WITHOUT_CLIENTLIBS)
+SET(MAN1_SERVER_CLIENTLIBS )
+ENDIF(NOT WITHOUT_CLIENTLIBS)
+SET(MAN1_SERVER innochecksum.1 ${MAN1_SERVER_CLIENTLIBS} myisam_ftdump.1 myisamchk.1
                 aria_chk.1 aria_dump_log.1 aria_ftdump.1 aria_pack.1 aria_read_log.1
 		myisamlog.1 myisampack.1 mysql.server.1
 		mysql_convert_table_format.1 mysql_fix_extensions.1
@@ -21,7 +26,7 @@
 		mysql_secure_installation.1 mysql_setpermission.1
 		mysql_tzinfo_to_sql.1 mysql_upgrade.1
 		mysqld_multi.1 mysqld_safe.1 mysqldumpslow.1 mysqlhotcopy.1
-		mysqltest.1 perror.1 replace.1 resolve_stack_dump.1
+		mysqltest.1 replace.1 resolve_stack_dump.1
 		resolveip.1 mariadb-service-convert.1
 		mysqld_safe_helper.1 tokuftdump.1 wsrep_sst_common.1
 		wsrep_sst_mysqldump.1 wsrep_sst_rsync.1
