--- a/usr/share/applications/teams.desktop	2019-11-20 09:24:54.000000000 -0600
+++ b/usr/share/applications/teams.desktop	2019-12-15 11:33:57.431428592 -0600
@@ -3,8 +3,8 @@
 Type=Application
 Name=Microsoft Teams - Preview
 Comment=Microsoft Teams for Linux is your chat-centered workspace in Office 365.
-Exec=/usr/bin/teams %U
-Icon=teams
+Exec=/opt/teams/bin/teams %U
+Icon=/opt/teams/share/pixmaps/teams.png
 Terminal=false
 StartupNotify=true
 StartupWMClass=Microsoft Teams - Preview
