From e73fe69f76b8166cb7149d085f69196d6a06d90e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Tue, 14 Jul 2020 18:05:32 +0200
Subject: [PATCH 13/22] Skip RDS socket tests that hang frequently

---
 Lib/test/test_socket.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Lib/test/test_socket.py b/Lib/test/test_socket.py
index eada6e929f..f63e509494 100644
--- a/Lib/test/test_socket.py
+++ b/Lib/test/test_socket.py
@@ -2084,6 +2084,7 @@ class BasicRDSTest(unittest.TestCase):
 
 
 @unittest.skipUnless(HAVE_SOCKET_RDS, 'RDS sockets required for this test.')
+@unittest.skip('Hangs frequently')
 class RDSTest(ThreadedRDSSocketTest):
 
     def __init__(self, methodName='runTest'):
-- 
2.35.1

