diff -Nuar NetworkManager-1.48.0/src/libnm-systemd-core/src/libsystemd-network/sd-dhcp-duid.c NetworkManager-1.48.0-fixed/src/libnm-systemd-core/src/libsystemd-network/sd-dhcp-duid.c
--- NetworkManager-1.48.0/src/libnm-systemd-core/src/libsystemd-network/sd-dhcp-duid.c	2024-05-31 11:23:57.000000000 -0000
+++ NetworkManager-1.48.0-fixed/src/libnm-systemd-core/src/libsystemd-network/sd-dhcp-duid.c	2024-06-09 19:50:50.492202803 -0000
@@ -193,6 +193,7 @@
         return 0;
 }
 
+#if 0
 int sd_dhcp_duid_set_uuid(sd_dhcp_duid *duid) {
         sd_id128_t machine_id;
         int r;
@@ -209,6 +210,7 @@
         duid->size = offsetof(struct duid, uuid.uuid) + sizeof(machine_id);
         return 0;
 }
+#endif
 
 int dhcp_duid_to_string_internal(uint16_t type, const void *data, size_t data_size, char **ret) {
         _cleanup_free_ char *p = NULL, *x = NULL;
diff -Nuar NetworkManager-1.48.0/src/libnm-systemd-core/src/libsystemd-network/sd-dhcp6-client.c NetworkManager-1.48.0-fixed/src/libnm-systemd-core/src/libsystemd-network/sd-dhcp6-client.c
--- NetworkManager-1.48.0/src/libnm-systemd-core/src/libsystemd-network/sd-dhcp6-client.c	2024-05-31 11:23:57.000000000 -0000
+++ NetworkManager-1.48.0-fixed/src/libnm-systemd-core/src/libsystemd-network/sd-dhcp6-client.c	2024-06-09 19:52:23.060726990 -0000
@@ -244,6 +244,7 @@
         return 0;
 }
 
+#if 0
 int sd_dhcp6_client_set_duid_uuid(sd_dhcp6_client *client) {
         int r;
 
@@ -256,6 +257,7 @@
 
         return 0;
 }
+#endif
 
 int sd_dhcp6_client_set_duid_raw(sd_dhcp6_client *client, uint16_t duid_type, const uint8_t *duid, size_t duid_len) {
         int r;
