From 0b0aa34bf1eb9c029b25ec59003673c4db107620 Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Wed, 4 Jul 2018 11:39:21 -0400
Subject: [PATCH 07/22] Disable nis

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index ee776b9d55..d5b1fddab9 100644
--- a/setup.py
+++ b/setup.py
@@ -1701,7 +1701,6 @@ class PyBuildExt(build_ext):
         self.detect_dbm_gdbm()
         self.detect_sqlite()
         self.detect_platform_specific_exts()
-        self.detect_nis()
         self.detect_compress_exts()
         self.detect_expat_elementtree()
         self.detect_multibytecodecs()
-- 
2.35.1

