diff -Nuar corebird-1.5/src/MainWindow.vala corebird-1.5-fixed/src/MainWindow.vala
--- corebird-1.5/src/MainWindow.vala	2017-05-02 13:34:40.000000000 +0000
+++ corebird-1.5-fixed/src/MainWindow.vala	2017-07-06 03:29:40.607003225 +0000
@@ -86,13 +86,7 @@
           continue;
       var e = new UserListEntry.from_account (acc);
       e.show_settings = true;
-      e.action_clicked.connect (() => {
-#if GTK322
-        account_popover.popdown ();
-#else
-        account_popover.hide ();
-#endif
-      });
+	  e.action_clicked.connect (() => { account_popover.hide ();});
       account_list.add (e);
     }
 
@@ -105,13 +99,7 @@
 
       var ule = new UserListEntry.from_account (new_acc);
       ule.show_settings = true;
-      ule.action_clicked.connect (() => {
-#if GTK322
-        account_popover.popdown ();
-#else
-        account_popover.hide ();
-#endif
-      });
+	  ule.action_clicked.connect (() => { account_popover.hide ();});
       account_list.add (ule);
       ule.show ();
     });
@@ -232,11 +220,6 @@
   [GtkCallback]
   private void account_row_activated_cb (Gtk.ListBoxRow row) {
     if (row is AddListEntry) {
-#if GTK322
-      account_popover.popdown ();
-#else
-      account_popover.hide ();
-#endif
       Account dummy_acc = new Account (0, Account.DUMMY, "name");
       var window = new MainWindow (application, dummy_acc);
       get_application ().add_window (window);
@@ -250,11 +233,6 @@
     MainWindow? account_window = null;
     if (user_id == this.account.id ||
         cb.is_window_open_for_user_id (user_id, out account_window)) {
-#if GTK322
-      account_popover.popdown ();
-#else
-      account_popover.hide ();
-#endif
 
       if (account_window != null)
         account_window.present ();
@@ -265,11 +243,6 @@
     Account? acc = Account.query_account_by_id (user_id);
     if (acc != null) {
       change_account (acc);
-#if GTK322
-      account_popover.popdown ();
-#else
-      account_popover.hide ();
-#endif
     } else
       warning ("account == null");
   }
@@ -351,11 +324,6 @@
   /* for show-account-list GAction */
   private void show_account_list () {
     if (this.account != null && this.account.screen_name != Account.DUMMY) {
-#if GTK322
-      this.account_popover.popup ();
-#else
-      this.account_popover.show ();
-#endif
     }
   }
 
@@ -365,20 +333,13 @@
 
   [GtkCallback]
   private void account_button_clicked_cb () {
-#if GTK322
-    if (account_popover.visible) {
-      account_popover.popdown ();
-    } else {
-      account_popover.popup ();
-    }
-#else
     account_popover.visible = !account_popover.visible;
-#endif
   }
 
   [GtkCallback]
   private void account_popover_closed_cb () {
     account_button.active = false;
+	account_popover.hide ();
   }
 
   [GtkCallback]
