From a17875321cefa50efd8b0146d7ee2c8f42d26613 Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Wed, 4 Jul 2018 11:39:21 -0400
Subject: [PATCH 07/19] Disable nis

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index dcca74c217..712d67e2f9 100644
--- a/setup.py
+++ b/setup.py
@@ -1388,7 +1388,7 @@ class PyBuildExt(build_ext):
         else:
             missing.extend(['resource', 'termios'])
 
-        nis = self._detect_nis(inc_dirs, lib_dirs)
+        nis = None
         if nis is not None:
             exts.append(nis)
         else:
-- 
2.35.1

