diff -Nuar ftgl-2.1.3~rc5/configure.ac ftgl-2.1.3~rc5-fixed/configure.ac
--- ftgl-2.1.3~rc5/configure.ac	2008-06-12 13:14:43.000000000 +0000
+++ ftgl-2.1.3~rc5-fixed/configure.ac	2017-12-20 16:24:36.033637972 +0000
@@ -50,8 +50,7 @@
 FTGL_CHECK_GLUT
 FTGL_CHECK_FONT
 
-PKG_CHECK_MODULES(CPPUNIT, cppunit, [CPPUNIT="yes"], [CPPUNIT="no"])
-AC_MSG_RESULT($CPPUNIT)
+CPPUNIT="no"
 AM_CONDITIONAL(HAVE_CPPUNIT, test "x$CPPUNIT" != "xno")
 
 dnl search the include directory (required for non-srcdir builds).

