diff -Nuar libgda-5.2.4/libgda/gda-column.h libgda-5.2.4-fixed/libgda/gda-column.h
--- libgda-5.2.4/libgda/gda-column.h	2013-10-19 15:35:15.000000000 +0300
+++ libgda-5.2.4-fixed/libgda/gda-column.h	2018-10-09 19:00:33.949046671 +0300
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2005 - 2011 Vivien Malerba <malerba@gnome-db.org>
- * Copyright (C) 2005 ¡lvaro PeÒa <alvaropg@telefonica.net>
+ * Copyright (C) 2005 √Ålvaro Pe√±a <alvaropg@telefonica.net>
  * Copyright (C) 2008 Przemys≈Çaw Grzegorczyk <pgrzegorczyk@gmail.com>
  *
  * This library is free software; you can redistribute it and/or
diff -Nuar libgda-5.2.4/libgda/gda-connection.h libgda-5.2.4-fixed/libgda/gda-connection.h
--- libgda-5.2.4/libgda/gda-connection.h	2015-06-13 11:36:25.000000000 +0300
+++ libgda-5.2.4-fixed/libgda/gda-connection.h	2018-10-09 19:08:38.501058187 +0300
@@ -6,7 +6,7 @@
  * Copyright (C) 2002 - 2003 Gonzalo Paniagua Javier <gonzalo@ximian.com>
  * Copyright (C) 2003 Filip Van Raemdonck <mechanix@debian.org>
  * Copyright (C) 2004 - 2005 Alan Knowles <alank@src.gnome.org>
- * Copyright (C) 2004 JosÈ MarÌa Casanova Crespo <jmcasanova@igalia.com>
+ * Copyright (C) 2004 Jos√© Mar√≠a Casanova Crespo <jmcasanova@igalia.com>
  * Copyright (C) 2005 - 2009 Bas Driessen <bas.driessen@xobas.com>
  * Copyright (C) 2006 - 2008 Murray Cumming <murrayc@murrayc.com>
  * Copyright (C) 2007 Leonardo Boshell <lb@kmc.com.co>
diff -Nuar libgda-5.2.4/libgda/gda-data-model-extra.h libgda-5.2.4-fixed/libgda/gda-data-model-extra.h
--- libgda-5.2.4/libgda/gda-data-model-extra.h	2015-06-12 23:05:14.000000000 +0300
+++ libgda-5.2.4-fixed/libgda/gda-data-model-extra.h	2018-10-09 19:02:01.491048751 +0300
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2001 Carlos PerellÛ MarÌn <carlos@gnome-db.org>
+ * Copyright (C) 2001 Carlos Perell√≥ Mar√≠n <carlos@gnome-db.org>
  * Copyright (C) 2001 - 2003 Rodrigo Moya <rodrigo@gnome-db.org>
  * Copyright (C) 2001 - 2011 Vivien Malerba <malerba@gnome-db.org>
  * Copyright (C) 2002 - 2003 Gonzalo Paniagua Javier <gonzalo@gnome-db.org>
diff -Nuar libgda-5.2.4/libgda/gda-data-model-iter-extra.h libgda-5.2.4-fixed/libgda/gda-data-model-iter-extra.h
--- libgda-5.2.4/libgda/gda-data-model-iter-extra.h	2015-06-12 23:05:14.000000000 +0300
+++ libgda-5.2.4-fixed/libgda/gda-data-model-iter-extra.h	2018-10-09 19:02:43.887049759 +0300
@@ -1,7 +1,7 @@
 /*
  * Copyright (C) 2000 Reinhard M√ºller <reinhard@src.gnome.org>
  * Copyright (C) 2000 - 2002 Rodrigo Moya <rodrigo@gnome-db.org>
- * Copyright (C) 2001 Carlos PerellÛ MarÌn <carlos@gnome-db.org>
+ * Copyright (C) 2001 Carlos Perell√≥ Mar√≠n <carlos@gnome-db.org>
  * Copyright (C) 2001 - 2011 Vivien Malerba <malerba@gnome-db.org>
  * Copyright (C) 2002 Gonzalo Paniagua Javier <gonzalo@src.gnome.org>
  *
diff -Nuar libgda-5.2.4/libgda/gda-data-model.h libgda-5.2.4-fixed/libgda/gda-data-model.h
--- libgda-5.2.4/libgda/gda-data-model.h	2015-06-13 11:36:25.000000000 +0300
+++ libgda-5.2.4-fixed/libgda/gda-data-model.h	2018-10-09 19:03:22.629050679 +0300
@@ -4,7 +4,7 @@
  * Copyright (C) 2003 Laurent Sansonetti <lrz@gnome.org>
  * Copyright (C) 2005 Dan Winship <danw@src.gnome.org>
  * Copyright (C) 2005 - 2012 Vivien Malerba <malerba@gnome-db.org>
- * Copyright (C) 2005 ¡lvaro PeÒa <alvaropg@telefonica.net>
+ * Copyright (C) 2005 √Ålvaro Pe√±a <alvaropg@telefonica.net>
  * Copyright (C) 2007 Murray Cumming <murrayc@murrayc.com>
  * Copyright (C) 2011 Daniel Espinosa <despinosa@src.gnome.org>
  *
diff -Nuar libgda-5.2.4/libgda/gda-mutex.h libgda-5.2.4-fixed/libgda/gda-mutex.h
--- libgda-5.2.4/libgda/gda-mutex.h	2015-06-13 11:36:25.000000000 +0300
+++ libgda-5.2.4-fixed/libgda/gda-mutex.h	2018-10-09 19:04:02.905051637 +0300
@@ -1,7 +1,7 @@
 /*
  * Copyright (C) 2000 Reinhard M√ºller <reinhard@src.gnome.org>
  * Copyright (C) 2000 - 2002 Rodrigo Moya <rodrigo@gnome-db.org>
- * Copyright (C) 2001 Carlos PerellÛ MarÌn <carlos@gnome-db.org>
+ * Copyright (C) 2001 Carlos Perell√≥ Mar√≠n <carlos@gnome-db.org>
  * Copyright (C) 2001 - 2013 Vivien Malerba <malerba@gnome-db.org>
  * Copyright (C) 2002 Gonzalo Paniagua Javier <gonzalo@src.gnome.org>
  *
diff -Nuar libgda-5.2.4/libgda/gda-row.h libgda-5.2.4-fixed/libgda/gda-row.h
--- libgda-5.2.4/libgda/gda-row.h	2013-10-19 15:35:15.000000000 +0300
+++ libgda-5.2.4-fixed/libgda/gda-row.h	2018-10-09 19:05:12.887053300 +0300
@@ -3,10 +3,10 @@
  * Copyright (C) 2001 - 2011 Vivien Malerba <malerba@gnome-db.org>
  * Copyright (C) 2002 Gonzalo Paniagua Javier <gonzalo@gnome-db.org>
  * Copyright (C) 2003 Laurent Sansonetti <laurent@datarescue.be>
- * Copyright (C) 2003 Xabier RodrÌguez Calvar <xrcalvar@igalia.com>
+ * Copyright (C) 2003 Xabier Rodr√≠guez Calvar <xrcalvar@igalia.com>
  * Copyright (C) 2004 Paisa  Seeluangsawat <paisa@users.sf.net>
  * Copyright (C) 2005 Bas Driessen <bas.driessen@xobas.com>
- * Copyright (C) 2005 ¡lvaro PeÒa <alvaropg@telefonica.net>
+ * Copyright (C) 2005 √Ålvaro Pe√±a <alvaropg@telefonica.net>
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
diff -Nuar libgda-5.2.4/libgda/gda-server-provider-private.h libgda-5.2.4-fixed/libgda/gda-server-provider-private.h
--- libgda-5.2.4/libgda/gda-server-provider-private.h	2015-06-13 11:36:25.000000000 +0300
+++ libgda-5.2.4-fixed/libgda/gda-server-provider-private.h	2018-10-09 19:05:44.666054055 +0300
@@ -1,7 +1,7 @@
 /*
  * Copyright (C) 2005 Dan Winship <danw@src.gnome.org>
  * Copyright (C) 2005 - 2011 Vivien Malerba <malerba@gnome-db.org>
- * Copyright (C) 2005 ¡lvaro PeÒa <alvaropg@telefonica.net>
+ * Copyright (C) 2005 √Ålvaro Pe√±a <alvaropg@telefonica.net>
  * Copyright (C) 2007 Murray Cumming <murrayc@murrayc.com>
  *
  * This library is free software; you can redistribute it and/or
diff -Nuar libgda-5.2.4/libgda/gda-util.h libgda-5.2.4-fixed/libgda/gda-util.h
--- libgda-5.2.4/libgda/gda-util.h	2014-02-05 22:28:20.000000000 +0200
+++ libgda-5.2.4-fixed/libgda/gda-util.h	2018-10-09 19:06:33.030055205 +0300
@@ -1,7 +1,7 @@
 /*
  * Copyright (C) 2000 Reinhard M√ºller <reinhard@src.gnome.org>
  * Copyright (C) 2000 - 2002 Rodrigo Moya <rodrigo@gnome-db.org>
- * Copyright (C) 2001 Carlos PerellÛ MarÌn <carlos@gnome-db.org>
+ * Copyright (C) 2001 Carlos Perell√≥ Mar√≠n <carlos@gnome-db.org>
  * Copyright (C) 2001 - 2013 Vivien Malerba <malerba@gnome-db.org>
  * Copyright (C) 2002 Gonzalo Paniagua Javier <gonzalo@src.gnome.org>
  * Copyright (C) 2006 - 2007 Murray Cumming <murrayc@murrayc.com>
diff -Nuar libgda-5.2.4/libgda/libgda-global-variables.h libgda-5.2.4-fixed/libgda/libgda-global-variables.h
--- libgda-5.2.4/libgda/libgda-global-variables.h	2015-06-12 23:05:14.000000000 +0300
+++ libgda-5.2.4-fixed/libgda/libgda-global-variables.h	2018-10-09 19:07:00.671055862 +0300
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2001 Carlos PerellÛ MarÌn <carlos@gnome-db.org>
+ * Copyright (C) 2001 Carlos Perell√≥ Mar√≠n <carlos@gnome-db.org>
  * Copyright (C) 2001 - 2003 Rodrigo Moya <rodrigo@gnome-db.org>
  * Copyright (C) 2001 - 2011 Vivien Malerba <malerba@gnome-db.org>
  * Copyright (C) 2002 - 2003 Gonzalo Paniagua Javier <gonzalo@gnome-db.org>
diff -Nuar libgda-5.2.4/libgda/sql-parser/gda-sql-statement.h libgda-5.2.4-fixed/libgda/sql-parser/gda-sql-statement.h
--- libgda-5.2.4/libgda/sql-parser/gda-sql-statement.h	2015-06-12 23:05:14.000000000 +0300
+++ libgda-5.2.4-fixed/libgda/sql-parser/gda-sql-statement.h	2018-10-09 19:10:19.673060591 +0300
@@ -1,7 +1,7 @@
 /*
  * Copyright (C) 2000 Reinhard M√ºller <reinhard@src.gnome.org>
  * Copyright (C) 2000 - 2002 Rodrigo Moya <rodrigo@gnome-db.org>
- * Copyright (C) 2001 Carlos PerellÛ MarÌn <carlos@gnome-db.org>
+ * Copyright (C) 2001 Carlos Perell√≥ Mar√≠n <carlos@gnome-db.org>
  * Copyright (C) 2001 - 2011 Vivien Malerba <malerba@gnome-db.org>
  * Copyright (C) 2002 Gonzalo Paniagua Javier <gonzalo@src.gnome.org>
  * Copyright (C) 2011 Murray Cumming <murrayc@murrayc.com>
diff -Nuar libgda-5.2.4/libgda/sql-parser/gda-statement-struct-delete.h libgda-5.2.4-fixed/libgda/sql-parser/gda-statement-struct-delete.h
--- libgda-5.2.4/libgda/sql-parser/gda-statement-struct-delete.h	2013-10-19 15:35:16.000000000 +0300
+++ libgda-5.2.4-fixed/libgda/sql-parser/gda-statement-struct-delete.h	2018-10-09 19:11:01.218061579 +0300
@@ -1,7 +1,7 @@
 /*
  * Copyright (C) 2005 Dan Winship <danw@src.gnome.org>
  * Copyright (C) 2005 - 2011 Vivien Malerba <malerba@gnome-db.org>
- * Copyright (C) 2005 ¡lvaro PeÒa <alvaropg@telefonica.net>
+ * Copyright (C) 2005 √Ålvaro Pe√±a <alvaropg@telefonica.net>
  * Copyright (C) 2007 Armin Burgmeier <armin@openismus.com>
  * Copyright (C) 2007 - 2009 Murray Cumming <murrayc@murrayc.com>
  *
diff -Nuar libgda-5.2.4/libgda/sql-parser/gda-statement-struct-unknown.h libgda-5.2.4-fixed/libgda/sql-parser/gda-statement-struct-unknown.h
--- libgda-5.2.4/libgda/sql-parser/gda-statement-struct-unknown.h	2013-10-19 15:35:16.000000000 +0300
+++ libgda-5.2.4-fixed/libgda/sql-parser/gda-statement-struct-unknown.h	2018-10-09 19:11:48.036062691 +0300
@@ -1,7 +1,7 @@
 /*
  * Copyright (C) 2000 Reinhard M√ºller <reinhard@src.gnome.org>
  * Copyright (C) 2000 - 2002 Rodrigo Moya <rodrigo@gnome-db.org>
- * Copyright (C) 2001 Carlos PerellÛ MarÌn <carlos@gnome-db.org>
+ * Copyright (C) 2001 Carlos Perell√≥ Mar√≠n <carlos@gnome-db.org>
  * Copyright (C) 2001 - 2011 Vivien Malerba <malerba@gnome-db.org>
  * Copyright (C) 2002 Gonzalo Paniagua Javier <gonzalo@src.gnome.org>
  * Copyright (C) 2009 Murray Cumming <murrayc@murrayc.com>
diff -Nuar libgda-5.2.4/libgda/sql-parser/gda-statement-struct-util.h libgda-5.2.4-fixed/libgda/sql-parser/gda-statement-struct-util.h
--- libgda-5.2.4/libgda/sql-parser/gda-statement-struct-util.h	2013-10-19 15:35:16.000000000 +0300
+++ libgda-5.2.4-fixed/libgda/sql-parser/gda-statement-struct-util.h	2018-10-09 19:12:23.616063537 +0300
@@ -1,7 +1,7 @@
 /*
  * Copyright (C) 2005 Dan Winship <danw@src.gnome.org>
  * Copyright (C) 2005 - 2011 Vivien Malerba <malerba@gnome-db.org>
- * Copyright (C) 2005 ¡lvaro PeÒa <alvaropg@telefonica.net>
+ * Copyright (C) 2005 √Ålvaro Pe√±a <alvaropg@telefonica.net>
  * Copyright (C) 2007 - 2009 Murray Cumming <murrayc@murrayc.com>
  *
  * This library is free software; you can redistribute it and/or
diff -Nuar libgda-5.2.4/libgda-ui/gdaui-easy.h libgda-5.2.4-fixed/libgda-ui/gdaui-easy.h
--- libgda-5.2.4/libgda-ui/gdaui-easy.h	2013-10-19 15:35:28.000000000 +0300
+++ libgda-5.2.4-fixed/libgda-ui/gdaui-easy.h	2018-10-09 19:07:47.148056966 +0300
@@ -1,7 +1,7 @@
 /*
  * Copyright (C) 2000 Reinhard M√ºller <reinhard@src.gnome.org>
  * Copyright (C) 2000 - 2002 Rodrigo Moya <rodrigo@gnome-db.org>
- * Copyright (C) 2001 Carlos PerellÛ MarÌn <carlos@gnome-db.org>
+ * Copyright (C) 2001 Carlos Perell√≥ Mar√≠n <carlos@gnome-db.org>
  * Copyright (C) 2001 - 2011 Vivien Malerba <malerba@gnome-db.org>
  * Copyright (C) 2002 Gonzalo Paniagua Javier <gonzalo@src.gnome.org>
  *
