Index: src/3rdparty/chromium/media/gpu/vaapi/vaapi_wrapper.cc
===================================================================
--- a/src/3rdparty/chromium/media/gpu/vaapi/vaapi_wrapper.cc
+++ b/src/3rdparty/chromium/media/gpu/vaapi/vaapi_wrapper.cc
@@ -400,6 +400,8 @@ static VAEntrypoint GetVaEntryPoint(Vaap
     case VaapiWrapper::kCodecModeMax:
       NOTREACHED();
       return VAEntrypointVLD;
+    default:
+      return 0;
   }
 }
 
Index: src/3rdparty/chromium/cc/input/snap_fling_controller.cc
===================================================================
--- a/src/3rdparty/chromium/cc/input/snap_fling_controller.cc
+++ b/src/3rdparty/chromium/cc/input/snap_fling_controller.cc
@@ -26,6 +26,8 @@ bool SnapFlingController::FilterEventFor
     case GestureScrollType::kEnd: {
       return state_ == State::kActive || state_ == State::kFinished;
     }
+    default:
+      return false;
   }
 }
 
Index: src/3rdparty/chromium/chrome/browser/ui/webui/discards/discards_ui.cc
===================================================================
--- a/src/3rdparty/chromium/chrome/browser/ui/webui/discards/discards_ui.cc
+++ b/src/3rdparty/chromium/chrome/browser/ui/webui/discards/discards_ui.cc
@@ -48,6 +48,8 @@ mojom::LifecycleUnitVisibility GetLifecy
       return mojom::LifecycleUnitVisibility::OCCLUDED;
     case content::Visibility::VISIBLE:
       return mojom::LifecycleUnitVisibility::VISIBLE;
+    default:
+      return mojom::LifecycleUnitVisibility::VISIBLE;
   }
 #if defined(COMPILER_MSVC)
   NOTREACHED();
Index: src/3rdparty/chromium/components/autofill/core/browser/autofill_manager.cc
===================================================================
--- a/src/3rdparty/chromium/components/autofill/core/browser/autofill_manager.cc
+++ b/src/3rdparty/chromium/components/autofill/core/browser/autofill_manager.cc
@@ -285,6 +285,7 @@ PopupType AutofillManager::GetPopupType(
 
     default:
       NOTREACHED();
+      return PopupType::kUnspecified;
   }
 }
 
Index: src/3rdparty/chromium/components/cast_channel/cast_message_util.cc
===================================================================
--- a/src/3rdparty/chromium/components/cast_channel/cast_message_util.cc
+++ b/src/3rdparty/chromium/components/cast_channel/cast_message_util.cc
@@ -368,6 +368,8 @@ const char* GetAppAvailabilityResultToSt
       return "unavailable";
     case GetAppAvailabilityResult::kUnknown:
       return "unknown";
+    default:
+      return "unknown";
   }
 }
 
Index: src/3rdparty/chromium/components/data_reduction_proxy/core/browser/data_reduction_proxy_util.cc
===================================================================
--- a/src/3rdparty/chromium/components/data_reduction_proxy/core/browser/data_reduction_proxy_util.cc
+++ b/src/3rdparty/chromium/components/data_reduction_proxy/core/browser/data_reduction_proxy_util.cc
@@ -337,6 +337,8 @@ PageloadMetrics_ConnectionType ProtoConn
       return PageloadMetrics_ConnectionType_CONNECTION_NONE;
     case net::NetworkChangeNotifier::CONNECTION_BLUETOOTH:
       return PageloadMetrics_ConnectionType_CONNECTION_BLUETOOTH;
+    default:
+      return PageloadMetrics_ConnectionType_CONNECTION_UNKNOWN;
   }
 }
 
@@ -351,6 +353,8 @@ RequestInfo_Protocol ProtoRequestInfoPro
       return RequestInfo_Protocol_QUIC;
     case DataReductionProxyData::RequestInfo::Protocol::UNKNOWN:
       return RequestInfo_Protocol_UNKNOWN;
+    default:
+      return RequestInfo_Protocol_UNKNOWN;
   }
 }
 
Index: src/3rdparty/chromium/content/browser/cache_storage/cache_storage_quota_client.cc
===================================================================
--- a/src/3rdparty/chromium/content/browser/cache_storage/cache_storage_quota_client.cc
+++ b/src/3rdparty/chromium/content/browser/cache_storage/cache_storage_quota_client.cc
@@ -98,6 +98,8 @@ storage::QuotaClient::ID CacheStorageQuo
       return kServiceWorkerCache;
     case CacheStorageOwner::kBackgroundFetch:
       return kBackgroundFetch;
+    default:
+      return kBackgroundFetch;
   }
 }
 
Index: src/3rdparty/chromium/media/capture/video/video_capture_device_descriptor.cc
===================================================================
--- a/src/3rdparty/chromium/media/capture/video/video_capture_device_descriptor.cc
+++ b/src/3rdparty/chromium/media/capture/video/video_capture_device_descriptor.cc
@@ -94,6 +94,8 @@ const char* VideoCaptureDeviceDescriptor
       return "Virtual Device";
     case VideoCaptureApi::UNKNOWN:
       return "Unknown";
+    default:
+      return "Unknown";
   }
 }
 
Index: src/3rdparty/chromium/net/nqe/network_quality_estimator.cc
===================================================================
--- a/src/3rdparty/chromium/net/nqe/network_quality_estimator.cc
+++ b/src/3rdparty/chromium/net/nqe/network_quality_estimator.cc
@@ -1342,6 +1342,8 @@ base::TimeDelta NetworkQualityEstimator:
     case nqe::internal::OBSERVATION_CATEGORY_COUNT:
       NOTREACHED();
       return base::TimeDelta();
+    default:
+      return base::TimeDelta();
   }
 }
 
Index: src/3rdparty/chromium/services/audio/public/cpp/audio_system_to_service_adapter.cc
===================================================================
--- a/src/3rdparty/chromium/services/audio/public/cpp/audio_system_to_service_adapter.cc
+++ b/src/3rdparty/chromium/services/audio/public/cpp/audio_system_to_service_adapter.cc
@@ -66,6 +66,8 @@ const char* GetTraceEvent(Action action)
       return "AudioSystemToServiceAdapter::GetAssociatedOutputDeviceID";
     case kGetInputDeviceInfo:
       return "AudioSystemToServiceAdapter::GetInputDeviceInfo";
+    default:
+      return "Unknown";
   }
   NOTREACHED();
 }
Index: src/3rdparty/chromium/services/network/cross_origin_read_blocking.cc
===================================================================
--- a/src/3rdparty/chromium/services/network/cross_origin_read_blocking.cc
+++ b/src/3rdparty/chromium/services/network/cross_origin_read_blocking.cc
@@ -800,6 +800,8 @@ bool CrossOriginReadBlocking::ResponseAn
       return sniffers_.empty() && !found_blockable_content_;
     case kBlock:
       return false;
+    default:
+      return true;
   }
 }
 
@@ -811,6 +813,8 @@ bool CrossOriginReadBlocking::ResponseAn
       return sniffers_.empty() && found_blockable_content_;
     case kBlock:
       return true;
+    default:
+      return false;
   }
 }
 
Index: src/3rdparty/chromium/third_party/blink/renderer/modules/accessibility/ax_layout_object.cc
===================================================================
--- a/src/3rdparty/chromium/third_party/blink/renderer/modules/accessibility/ax_layout_object.cc
+++ b/src/3rdparty/chromium/third_party/blink/renderer/modules/accessibility/ax_layout_object.cc
@@ -1191,6 +1191,8 @@ AXTextPosition AXLayoutObject::GetTextPo
       return kAXTextPositionSubscript;
     case EVerticalAlign::kSuper:
       return kAXTextPositionSuperscript;
+    default:
+      return AXNodeObject::GetTextPosition();
   }
 }
 
Index: src/3rdparty/chromium/third_party/blink/renderer/modules/webaudio/audio_param.cc
===================================================================
--- a/src/3rdparty/chromium/third_party/blink/renderer/modules/webaudio/audio_param.cc
+++ b/src/3rdparty/chromium/third_party/blink/renderer/modules/webaudio/audio_param.cc
@@ -144,6 +144,7 @@ String AudioParamHandler::GetParamName()
       return custom_param_name_;
     default:
       NOTREACHED();
+      return "Unknown";
   }
 }
 
Index: src/3rdparty/chromium/ui/accessibility/platform/ax_platform_node_auralinux.cc
===================================================================
--- a/src/3rdparty/chromium/ui/accessibility/platform/ax_platform_node_auralinux.cc
+++ b/src/3rdparty/chromium/ui/accessibility/platform/ax_platform_node_auralinux.cc
@@ -1291,6 +1291,8 @@ AtkRole AXPlatformNodeAuraLinux::GetAtkR
     case ax::mojom::Role::kPresentational:
     case ax::mojom::Role::kUnknown:
       return ATK_ROLE_REDUNDANT_OBJECT;
+    default:
+      return ATK_ROLE_REDUNDANT_OBJECT;
   }
 }
 
Index: src/3rdparty/chromium/components/password_manager/core/browser/browser_save_password_progress_logger.cc
===================================================================
--- a/src/3rdparty/chromium/components/password_manager/core/browser/browser_save_password_progress_logger.cc
+++ b/src/3rdparty/chromium/components/password_manager/core/browser/browser_save_password_progress_logger.cc
@@ -86,6 +86,8 @@ std::string VoteTypeToString(
       return "HTML classifier";
     case AutofillUploadContents::Field::FIRST_USE:
       return "First use";
+    default:
+      return "No information";
   }
 }
 
Index: src/3rdparty/chromium/components/previews/content/previews_hints.cc
===================================================================
--- a/src/3rdparty/chromium/components/previews/content/previews_hints.cc
+++ b/src/3rdparty/chromium/components/previews/content/previews_hints.cc
@@ -94,6 +94,8 @@ ConvertProtoOptimizationTypeToPreviewsOp
       return PreviewsType::NOSCRIPT;
     case optimization_guide::proto::RESOURCE_LOADING:
       return PreviewsType::RESOURCE_LOADING_HINTS;
+    default:
+      return base::nullopt;
   }
 }
 
Index: src/3rdparty/chromium/components/previews/core/previews_black_list.cc
===================================================================
--- a/src/3rdparty/chromium/components/previews/core/previews_black_list.cc
+++ b/src/3rdparty/chromium/components/previews/core/previews_black_list.cc
@@ -33,6 +33,9 @@ PreviewsEligibilityReason BlacklistReaso
       return PreviewsEligibilityReason::ALLOWED;
     case blacklist::BlacklistReason::kAllowed:
       return PreviewsEligibilityReason::ALLOWED;
+    default:
+      NOTREACHED() << "Previews does not support type-base blacklisting";
+      return PreviewsEligibilityReason::ALLOWED;
   }
 }
 
Index: src/3rdparty/chromium/content/browser/frame_host/frame_tree_node.cc
===================================================================
--- a/src/3rdparty/chromium/content/browser/frame_host/frame_tree_node.cc
+++ b/src/3rdparty/chromium/content/browser/frame_host/frame_tree_node.cc
@@ -624,6 +624,7 @@ bool FrameTreeNode::UpdateUserActivation
       return NotifyUserActivation();
   }
   NOTREACHED() << "Invalid update_type.";
+  return false;
 }
 
 void FrameTreeNode::OnSetHasReceivedUserGestureBeforeNavigation(bool value) {
Index: src/3rdparty/chromium/content/renderer/input/input_event_prediction.cc
===================================================================
--- a/src/3rdparty/chromium/content/renderer/input/input_event_prediction.cc
+++ b/src/3rdparty/chromium/content/renderer/input/input_event_prediction.cc
@@ -76,6 +76,8 @@ std::unique_ptr<ui::InputPredictor> Inpu
       return std::make_unique<ui::LeastSquaresPredictor>();
     case PredictorType::kKalman:
       return std::make_unique<ui::KalmanPredictor>();
+    default:
+      return std::make_unique<ui::EmptyPredictor>();
   }
 }
 
Index: src/3rdparty/chromium/content/renderer/media/webrtc/peer_connection_tracker.cc
===================================================================
--- a/src/3rdparty/chromium/content/renderer/media/webrtc/peer_connection_tracker.cc
+++ b/src/3rdparty/chromium/content/renderer/media/webrtc/peer_connection_tracker.cc
@@ -112,6 +112,8 @@ static const char* SerializeDirection(
       return "'recvonly'";
     case webrtc::RtpTransceiverDirection::kInactive:
       return "'inactive'";
+    default:
+      return "'unknown'";
   }
 }
 
Index: src/3rdparty/chromium/device/bluetooth/bluetooth_remote_gatt_characteristic.cc
===================================================================
--- a/src/3rdparty/chromium/device/bluetooth/bluetooth_remote_gatt_characteristic.cc
+++ b/src/3rdparty/chromium/device/bluetooth/bluetooth_remote_gatt_characteristic.cc
@@ -397,6 +397,8 @@ bool BluetoothRemoteGattCharacteristic::
       return hasNotify;
     case NotificationType::kIndication:
       return hasIndicate;
+    default:
+      return false;
   }
 }
 
Index: src/3rdparty/chromium/media/blink/watch_time_reporter.cc
===================================================================
--- a/src/3rdparty/chromium/media/blink/watch_time_reporter.cc
+++ b/src/3rdparty/chromium/media/blink/watch_time_reporter.cc
@@ -571,6 +571,8 @@ WatchTimeKey WatchTimeReporter::GetDispl
       return DISPLAY_TYPE_KEY(DisplayFullscreen);
     case DisplayType::kPictureInPicture:
       return DISPLAY_TYPE_KEY(DisplayPictureInPicture);
+    default:
+      return DISPLAY_TYPE_KEY(DisplayInline);
   }
 }
 
Index: src/3rdparty/chromium/services/ui/public/cpp/gpu/command_buffer_metrics.cc
===================================================================
--- a/src/3rdparty/chromium/services/ui/public/cpp/gpu/command_buffer_metrics.cc
+++ b/src/3rdparty/chromium/services/ui/public/cpp/gpu/command_buffer_metrics.cc
@@ -86,6 +86,8 @@ std::string ContextTypeToString(ContextT
       return "Unknown";
     case ContextType::FOR_TESTING:
       return "ForTesting";
+    default:
+      return "Unknown";
   }
 }
 
Index: src/3rdparty/chromium/third_party/blink/renderer/core/css/style_environment_variables.cc
===================================================================
--- a/src/3rdparty/chromium/third_party/blink/renderer/core/css/style_environment_variables.cc
+++ b/src/3rdparty/chromium/third_party/blink/renderer/core/css/style_environment_variables.cc
@@ -67,6 +67,7 @@ const AtomicString StyleEnvironmentVaria
   }
 
   NOTREACHED();
+  return "unknown";
 }
 
 // static
Index: src/3rdparty/chromium/third_party/blink/renderer/core/html/media/media_controls.cc
===================================================================
--- a/src/3rdparty/chromium/third_party/blink/renderer/core/html/media/media_controls.cc
+++ b/src/3rdparty/chromium/third_party/blink/renderer/core/html/media/media_controls.cc
@@ -36,6 +36,9 @@ AtomicString MediaControls::GetSizingCSS
       return kMediaControlsSizingMediumCSSClass;
     case MediaControlsSizingClass::kLarge:
       return kMediaControlsSizingLargeCSSClass;
+    default:
+      NOTREACHED();
+      return kMediaControlsSizingMediumCSSClass;
   }
 
   NOTREACHED();
Index: src/3rdparty/chromium/third_party/blink/renderer/modules/cookie_store/cookie_change_event.cc
===================================================================
--- a/src/3rdparty/chromium/third_party/blink/renderer/modules/cookie_store/cookie_change_event.cc
+++ b/src/3rdparty/chromium/third_party/blink/renderer/modules/cookie_store/cookie_change_event.cc
@@ -53,6 +53,9 @@ String ToCookieListItemSameSite(network:
       return "lax";
     case network::mojom::CookieSameSite::NO_RESTRICTION:
       return "unrestricted";
+    default:
+      NOTREACHED();
+      return "unknown";
   }
 
   NOTREACHED();
Index: src/3rdparty/chromium/third_party/blink/renderer/platform/heap/heap_stats_collector.h
===================================================================
--- a/src/3rdparty/chromium/third_party/blink/renderer/platform/heap/heap_stats_collector.h
+++ b/src/3rdparty/chromium/third_party/blink/renderer/platform/heap/heap_stats_collector.h
@@ -94,6 +94,8 @@ class PLATFORM_EXPORT ThreadHeapStatsCol
         return "BlinkGC.VisitPersistents";
       case Id::kVisitStackRoots:
         return "BlinkGC.VisitStackRoots";
+      default:
+        return "BlinkGC.Unknown";
     }
   }
 
Index: src/3rdparty/chromium/third_party/blink/renderer/platform/scheduler/main_thread/page_scheduler_impl.cc
===================================================================
--- a/src/3rdparty/chromium/third_party/blink/renderer/platform/scheduler/main_thread/page_scheduler_impl.cc
+++ b/src/3rdparty/chromium/third_party/blink/renderer/platform/scheduler/main_thread/page_scheduler_impl.cc
@@ -688,6 +688,9 @@ PageSchedulerImpl::PageLifecycleStateTra
           NOTREACHED();
           return base::nullopt;
       }
+    default:
+      NOTREACHED();
+      return base::nullopt;
   }
 }
 
Index: src/3rdparty/chromium/ui/base/mojo/clipboard_struct_traits.h
===================================================================
--- a/src/3rdparty/chromium/ui/base/mojo/clipboard_struct_traits.h
+++ b/src/3rdparty/chromium/ui/base/mojo/clipboard_struct_traits.h
@@ -20,6 +20,9 @@ struct EnumTraits<ui::mojom::ClipboardTy
         return ui::mojom::ClipboardType::SELECTION;
       case ui::CLIPBOARD_TYPE_DRAG:
         return ui::mojom::ClipboardType::DRAG;
+      default:
+        NOTREACHED();
+        return ui::mojom::ClipboardType::SELECTION;
     }
   }
 
@@ -34,6 +37,8 @@ struct EnumTraits<ui::mojom::ClipboardTy
       case ui::mojom::ClipboardType::DRAG:
         *out = ui::CLIPBOARD_TYPE_DRAG;
         return true;
+      default:
+        return false;
     }
     NOTREACHED();
     return false;
Index: src/3rdparty/chromium/ui/views/layout/layout_provider.cc
===================================================================
--- a/src/3rdparty/chromium/ui/views/layout/layout_provider.cc
+++ b/src/3rdparty/chromium/ui/views/layout/layout_provider.cc
@@ -156,6 +156,8 @@ int LayoutProvider::GetCornerRadiusMetri
       return is_touch ? 8 : 4;
     case EMPHASIS_MAXIMUM:
       return is_touch ? std::min(size.width(), size.height()) / 2 : 4;
+    default:
+      return 0;
   }
 }
 
Index: src/3rdparty/chromium/third_party/blink/renderer/modules/peerconnection/rtc_rtp_transceiver.cc
===================================================================
--- a/src/3rdparty/chromium/third_party/blink/renderer/modules/peerconnection/rtc_rtp_transceiver.cc
+++ b/src/3rdparty/chromium/third_party/blink/renderer/modules/peerconnection/rtc_rtp_transceiver.cc
@@ -28,6 +28,8 @@ String TransceiverDirectionToString(
       return "recvonly";
     case webrtc::RtpTransceiverDirection::kInactive:
       return "inactive";
+    default:
+      return "unknown";
   }
   NOTREACHED();
 }
Index: src/3rdparty/chromium/third_party/blink/renderer/core/loader/mixed_content_checker.cc
===================================================================
--- a/src/3rdparty/chromium/third_party/blink/renderer/core/loader/mixed_content_checker.cc
+++ b/src/3rdparty/chromium/third_party/blink/renderer/core/loader/mixed_content_checker.cc
@@ -237,6 +237,7 @@ bool MixedContentChecker::IsMixedContent
     const FetchClientSettingsObjectImpl& settings,
     const KURL& url) {
   switch (settings.GetHttpsState()) {
+    default:
     case HttpsState::kNone:
       return false;
 
Index: src/3rdparty/chromium/content/common/navigation_params.cc
===================================================================
--- a/src/3rdparty/chromium/content/common/navigation_params.cc
+++ b/src/3rdparty/chromium/content/common/navigation_params.cc
@@ -36,6 +36,8 @@ bool IsNavigationDownloadAllowed(Navigat
     case NavigationDownloadPolicy::kAllowOpenerCrossOrigin:
     case NavigationDownloadPolicy::kAllowOpenerCrossOriginNoGesture:
       return true;
+    default:
+      __builtin_unreachable();
   }
 }
 
