diff --git a/start-qgle.sh b/start-qgle.sh
index c9f0ffa..3f6103d 100644
--- a/gui/start-qgle.sh
+++ b/gui/start-qgle.sh
@@ -1,3 +1,5 @@
+#!/bin/sh
+
 # ***********************************************************************************
 # * QGLE - A Graphical Interface to GLE                                             *
 # * Copyright (C) 2006  A. S. Budden & J. Struyf                                    *
@@ -19,10 +21,10 @@
 # * Also add information on how to contact you by electronic and paper mail.        *
 # ***********************************************************************************
 
-#!/bin/sh
 LD_LIBRARY_PATH=../../build/lib/:$LD_LIBRARY_PATH
-#GLE_TOP=/usr/share/gle/4.0.13-cvs/
+GLE_TOP=/usr/share/gle
 export LD_LIBRARY_PATH
-#export GLE_TOP
+export GLE_TOP
 #./qgle
-../../build/bin/qgle
+# ../../build/bin/qgle
+exec /usr/bin/qgle.bin $*
