--- a/configure.ac	2019-12-27 23:01:15.638575914 -0800
+++ b/configure.ac	2019-12-27 23:01:29.002525893 -0800
@@ -908,7 +908,7 @@
 	
 	# Ensure that no deprecated glibmm symbols are introduced.
 	# lp:inkscape builds cleanly with this option at r10957
-	CPPFLAGS="-DGLIBMM_DISABLE_DEPRECATED $CPPFLAGS"
+	#CPPFLAGS="-DGLIBMM_DISABLE_DEPRECATED $CPPFLAGS"
 	
 	dnl Pango 1.32.4 uses a deprecated Glib symbol:
 	dnl   https://bugzilla.gnome.org/show_bug.cgi?id=689843
