diff -Nuar kalzium-18.08.3/CMakeLists.txt kalzium-18.08.3-fixed/CMakeLists.txt
--- kalzium-18.08.3/CMakeLists.txt	2018-11-06 00:09:41.000000000 +0000
+++ kalzium-18.08.3-fixed/CMakeLists.txt	2019-01-18 15:39:44.637718701 +0000
@@ -38,12 +38,12 @@
 include_directories(${AvogadroLibs_INCLUDE_DIRS})
 
 find_package(Qt5 REQUIRED COMPONENTS
-    Widgets
-    Script
     Core
     Gui
-    OpenGL
+    Script
     Svg
+    Widgets
+    Xml
 )
 
 find_package(KF5 REQUIRED COMPONENTS
@@ -91,6 +91,7 @@
 endif(MSVC)
 
 if (OPENBABEL2_FOUND AND AvogadroLibs_FOUND AND EIGEN3_FOUND)
+  find_package(Qt5OpenGL ${QT_MIN_VERSION} REQUIRED)
   find_package(KF5NewStuff REQUIRED)
   # avoid compilerwarnings about redefinitions
   # todo: use check_function_exits() ?
diff -Nuar kalzium-18.08.3/compoundviewer/CMakeLists.txt kalzium-18.08.3-fixed/compoundviewer/CMakeLists.txt
--- kalzium-18.08.3/compoundviewer/CMakeLists.txt	2018-08-27 05:40:51.000000000 +0000
+++ kalzium-18.08.3-fixed/compoundviewer/CMakeLists.txt	2019-01-18 15:41:42.558882692 +0000
@@ -1,7 +1,7 @@
 include_directories(
     ${CMAKE_CURRENT_BINARY_DIR}/..
     ${EIGEN3_INCLUDE_DIR}
-    ${OPENBABEL2_INCLUDE_DIR} ${OPENGL_INCLUDE_DIRS}
+	${OPENBABEL2_INCLUDE_DIR}
     ${AvogadroLibs_INCLUDE_DIRS}
     ${PROJECT_SOURCE_DIR}/compoundviewer
 )
@@ -22,6 +22,7 @@
     KF5::KDELibs4Support
     Qt5::OpenGL
     Qt5::Gui
+    Qt5::Widgets
     ${OPENBABEL2_LIBRARIES}
     AvogadroQtGui
     AvogadroQtOpenGL
diff -Nuar kalzium-18.08.3/libscience/CMakeLists.txt kalzium-18.08.3-fixed/libscience/CMakeLists.txt
--- kalzium-18.08.3/libscience/CMakeLists.txt	2018-08-27 05:40:51.000000000 +0000
+++ kalzium-18.08.3-fixed/libscience/CMakeLists.txt	2019-01-18 15:42:26.239313694 +0000
@@ -16,7 +16,7 @@
 
 add_library(science SHARED ${science_LIB_SRCS})
 
-target_link_libraries(science Qt5::OpenGL Qt5::Xml KF5::UnitConversion KF5::WidgetsAddons KF5::KDELibs4Support)
+target_link_libraries(science Qt5::Xml KF5::UnitConversion KF5::WidgetsAddons KF5::KDELibs4Support)
 set_target_properties(science PROPERTIES VERSION ${KALZIUMLIB_VERSION} SOVERSION ${KALZIUMLIB_SOVERSION} )
 install(TARGETS science  ${INSTALL_TARGETS_DEFAULT_ARGS})
