diff -Nuar gvfs-1.28.3/metadata/meta-daemon.c gvfs-1.28.3-fixed/metadata/meta-daemon.c
--- gvfs-1.28.3/metadata/meta-daemon.c	2016-07-13 06:54:30.000000000 +0000
+++ gvfs-1.28.3-fixed/metadata/meta-daemon.c	2018-01-22 16:29:13.606867301 +0000
@@ -27,6 +27,7 @@
 #include <glib/gstdio.h>
 #include <locale.h>
 #include <stdlib.h>
+#include <sys/sysmacros.h>
 #include "metatree.h"
 #include "gvfsdaemonprotocol.h"
 #include "metadata-dbus.h"
diff -Nuar gvfs-1.28.3/metadata/metatree.c gvfs-1.28.3-fixed/metadata/metatree.c
--- gvfs-1.28.3/metadata/metatree.c	2016-07-13 06:54:30.000000000 +0000
+++ gvfs-1.28.3-fixed/metadata/metatree.c	2018-01-22 16:31:23.074118979 +0000
@@ -1,4 +1,5 @@
 #include "config.h"
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/mman.h>
 
