diff --git a/support-files/CMakeLists.txt b/support-files/CMakeLists.txt
index f4cc5f26..d51bba1a 100644
--- a/support-files/CMakeLists.txt
+++ b/support-files/CMakeLists.txt
@@ -129,10 +129,6 @@ IF(UNIX AND NOT WITHOUT_SERVER)
                    ${CMAKE_CURRENT_BINARY_DIR}/mariadb.socket @ONLY)
     CONFIGURE_FILE(mariadb-extra.socket.in
                    ${CMAKE_CURRENT_BINARY_DIR}/mariadb-extra.socket @ONLY)
-    EXECUTE_PROCESS(
-    COMMAND ${CMAKE_COMMAND} -E create_symlink ./mariadb.service mysql.service
-    COMMAND ${CMAKE_COMMAND} -E create_symlink ./mariadb.service mysqld.service
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
 
     IF(WITH_WSREP)
       INSTALL(FILES use_galera_new_cluster.conf
@@ -140,8 +136,8 @@ IF(UNIX AND NOT WITHOUT_SERVER)
     ENDIF()
     INSTALL(FILES
             ${CMAKE_CURRENT_BINARY_DIR}/mariadb.service
-            ${CMAKE_CURRENT_BINARY_DIR}/mysql.service
-            ${CMAKE_CURRENT_BINARY_DIR}/mysqld.service
+            ${CMAKE_CURRENT_BINARY_DIR}/mariadb.socket
+            ${CMAKE_CURRENT_BINARY_DIR}/mariadb-extra.socket
             DESTINATION ${inst_location}/systemd COMPONENT SupportFiles)
 
     # @ in directory name broken between CMake version 2.8.12.2 and 3.3
@@ -157,8 +153,6 @@ IF(UNIX AND NOT WITHOUT_SERVER)
               ${CMAKE_CURRENT_BINARY_DIR}/mariadb@.service
               ${CMAKE_CURRENT_BINARY_DIR}/mariadb@.socket
               ${CMAKE_CURRENT_BINARY_DIR}/mariadb-extra@.socket
-              ${CMAKE_CURRENT_BINARY_DIR}/mysql.service
-              ${CMAKE_CURRENT_BINARY_DIR}/mysqld.service
               DESTINATION ${inst_location}/systemd COMPONENT SupportFiles)
     ENDIF()
 
@@ -167,8 +161,6 @@ IF(UNIX AND NOT WITHOUT_SERVER)
               ${CMAKE_CURRENT_BINARY_DIR}/mariadb.service
               ${CMAKE_CURRENT_BINARY_DIR}/mariadb.socket
               ${CMAKE_CURRENT_BINARY_DIR}/mariadb-extra.socket
-              ${CMAKE_CURRENT_BINARY_DIR}/mysql.service
-              ${CMAKE_CURRENT_BINARY_DIR}/mysqld.service
               DESTINATION ${INSTALL_SYSTEMD_UNITDIR} COMPONENT Server)
 
       # http://public.kitware.com/Bug/view.php?id=14782
