diff --git a/libmariadb/plugins/auth/CMakeLists.txt b/libmariadb/plugins/auth/CMakeLists.txt
index b66cf170..d4e42c67 100644
--- a/libmariadb/plugins/auth/CMakeLists.txt
+++ b/libmariadb/plugins/auth/CMakeLists.txt
@@ -117,7 +117,7 @@
     SET_SOURCE_FILES_PROPERTIES(${GSSAPI_SOURCES} PROPERTY COMPILE_FLAGS "-Wno-deprecated-declarations")
   ENDIF()
   IF(have_C__Wl___as_needed)
-    SET_TARGET_PROPERTIES(auth_gssapi_client PROPERTIES LINK_FLAGS "-Wl,--as-needed")
+    SET_SOURCE_FILES_PROPERTIES(${GSSAPI_SOURCES} PROPERTY LINK_FLAGS "-Wl,--as-needed")
   ENDIF()
 ENDIF()
 
