diff --git a/tool/rbinstall.rb b/tool/rbinstall.rb
index 79a96a7..e13e1af 100755
--- a/tool/rbinstall.rb
+++ b/tool/rbinstall.rb
@@ -1010,7 +1010,7 @@ def install_default_gem(dir, srcdir, bindir)
 end

 install?(:ext, :comm, :gem, :'bundled-gems') do
-  gem_dir = Gem.default_dir
+  gem_dir = ENV['GEM_DESTDIR']
   install_dir = with_destdir(gem_dir)
   prepare "bundled gems", gem_dir
   RbInstall.no_write do
