diff -Nuar qbittorrent-4.0.1/cmake/Modules/MacroQbtCompilerSettings.cmake qbittorrent-4.0.1-fixed/cmake/Modules/MacroQbtCompilerSettings.cmake
--- qbittorrent-4.0.1/cmake/Modules/MacroQbtCompilerSettings.cmake	2017-11-26 12:31:46.029909912 +0000
+++ qbittorrent-4.0.1-fixed/cmake/Modules/MacroQbtCompilerSettings.cmake	2017-11-26 12:53:45.103456112 +0000
@@ -13,16 +13,12 @@
             "-Wfloat-equal -Wcast-qual -Wcast-align"
             "-Wsign-conversion -Winvalid-pch -Werror=return-type -Wno-long-long"
 #			-fstack-protector-all
-            "-Werror -Wno-error=deprecated-declarations"
         )
         set (_GCC_COMMON_CXX_FLAGS  "-fexceptions -frtti"
             "-Woverloaded-virtual -Wold-style-cast -Wstrict-null-sentinel"
             "-Wnon-virtual-dtor -Wfloat-equal -Wcast-qual -Wcast-align"
             "-Werror=overloaded-virtual"
     # 		"-Weffc++"
-            "-Werror -Wno-error=cpp"
-            # we should modify code to make these ones obsolete
-            "-Wno-error=sign-conversion -Wno-error=float-equal"
         )
 
         if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.9)

