diff -Nuar wireshark-2.6.0/plugins/codecs/l16_mono/CMakeLists.txt wireshark-2.6.0-fixed/plugins/codecs/l16_mono/CMakeLists.txt
--- wireshark-2.6.0/plugins/codecs/l16_mono/CMakeLists.txt	2018-04-24 17:39:12.000000000 -0000
+++ wireshark-2.6.0-fixed/plugins/codecs/l16_mono/CMakeLists.txt	2018-05-14 16:28:19.082638789 -0000
@@ -28,7 +28,7 @@
 )
 
 include_directories(
-	${CMAKE_SOURCE_DIR}/codec
+	${CMAKE_SOURCE_DIR}/codecs
 	${CMAKE_CURRENT_SOURCE_DIR}
 )
 
diff -Nuar wireshark-2.6.0/plugins/codecs/l16_mono/Makefile.am wireshark-2.6.0-fixed/plugins/codecs/l16_mono/Makefile.am
--- wireshark-2.6.0/plugins/codecs/l16_mono/Makefile.am	2018-04-24 17:39:12.000000000 -0000
+++ wireshark-2.6.0-fixed/plugins/codecs/l16_mono/Makefile.am	2018-05-14 16:27:45.162897064 -0000
@@ -26,7 +26,7 @@
 NONGENERATED_C_FILES = \
 	$(NONGENERATED_REGISTER_C_FILES)
 
-wiretap_plugin_LTLIBRARIES = l16mono.la
+codecs_plugin_LTLIBRARIES = l16mono.la
 
 l16mono_la_SOURCES = \
 	$(SRC_FILES)	\
