diff --git a/setup.py b/setup.py
index 688e7703..e1ab9b65 100755
--- a/setup.py
+++ b/setup.py
@@ -995,6 +995,15 @@ def create_linux_bundle_gunk(ddir: str, libdir_name: str) -> None:
     os.makedirs(os.path.dirname(in_src_launcher), exist_ok=True)
     os.symlink(os.path.relpath(launcher, os.path.dirname(in_src_launcher)), in_src_launcher)
 
+def terminfo(args):
+    ddir = args.prefix
+    libdir = os.path.join(ddir, args.libdir_name.strip('/'), 'kitty')
+    build_terminfo = runpy.run_path('build-terminfo', run_name='import_build')
+    for x in (libdir, os.path.join(ddir, 'share')):
+        odir = os.path.join(x, 'terminfo')
+        safe_makedirs(odir)
+        build_terminfo['compile_terminfo'](odir)
+
 
 def macos_info_plist() -> bytes:
     import plistlib
@@ -1324,7 +1333,7 @@ def safe_remove(*entries: str) -> None:
 
     safe_remove(
         'build', 'compile_commands.json', 'link_commands.json',
-        'linux-package', 'kitty.app', 'asan-launcher',
+        'linux-terminfo', 'linux-package', 'kitty.app', 'asan-launcher',
         'kitty-profile', 'kitty/launcher')
 
     def excluded(root: str, d: str) -> bool:
@@ -1354,6 +1363,7 @@ def option_parser() -> argparse.ArgumentParser:  # {{{
         choices=('build',
                  'test',
                  'linux-package',
+                 'linux-terminfo',
                  'kitty.app',
                  'linux-freeze',
                  'macos-freeze',
@@ -1560,6 +1570,8 @@ def main() -> None:
         elif args.action == 'linux-package':
             build(args, native_optimizations=False)
             package(args, bundle_type='linux-package')
+        elif args.action == 'linux-terminfo':
+            terminfo(args)
         elif args.action == 'linux-freeze':
             build(args, native_optimizations=False)
             package(args, bundle_type='linux-freeze')
