From c0bd5493d7057cd423c12b794d34cad3d16ebb2a Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Fri, 6 Jul 2018 14:49:00 -0400
Subject: [PATCH 08/14] Install shared libs in LIBDEST

---
 Makefile.pre.in | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index 08a36919db..171946064d 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -147,7 +147,6 @@ SCRIPTDIR=	$(prefix)/$(PLATLIBDIR)
 ABIFLAGS=	@ABIFLAGS@
 
 # Detailed destination directories
-BINLIBDEST=	@BINLIBDEST@
 LIBDEST=	$(SCRIPTDIR)/python$(VERSION)
 INCLUDEPY=	$(INCLUDEDIR)/python$(LDVERSION)
 CONFINCLUDEPY=	$(CONFINCLUDEDIR)/python$(LDVERSION)
@@ -158,7 +157,7 @@ EXT_SUFFIX=	@EXT_SUFFIX@
 LDSHARED=	@LDSHARED@ $(PY_LDFLAGS)
 BLDSHARED=	@BLDSHARED@ $(PY_CORE_LDFLAGS)
 LDCXXSHARED=	@LDCXXSHARED@
-DESTSHARED=	$(BINLIBDEST)/lib-dynload
+DESTSHARED=	$(LIBDEST)/lib-dynload
 
 # List of exported symbols for AIX
 EXPORTSYMS=	@EXPORTSYMS@
@@ -1274,7 +1273,7 @@ commoninstall:  check-clean-src @FRAMEWORKALTINSTALLFIRST@ \
 		@FRAMEWORKALTINSTALLLAST@
 
 # Install shared libraries enabled by Setup
-DESTDIRS=	$(exec_prefix) $(LIBDIR) $(BINLIBDEST) $(DESTSHARED)
+DESTDIRS=	$(exec_prefix) $(LIBDIR) $(DESTSHARED)
 
 oldsharedinstall: $(DESTSHARED) $(SHAREDMODS)
 		@for i in X $(SHAREDMODS); do \
-- 
2.35.1

