diff -Nuar seahorse-3.20.0/configure.ac seahorse-3.20.0-fixed/configure.ac
--- seahorse-3.20.0/configure.ac	2016-03-25 08:10:13.000000000 +0000
+++ seahorse-3.20.0-fixed/configure.ac	2017-10-21 13:33:40.416325284 +0000
@@ -9,7 +9,7 @@
 GTK_REQ=3.4.0
 GTK_MAX=GTK_VERSION_3_4
 
-GNUPG_ACCEPTED="2.0.12 2.1.4"
+GNUPG_ACCEPTED="2.0.12 2.1.4 2.2.0"
 GPGME_REQUIRED=1.0.0
 LIBSECRET_REQUIRED=0.16
 AVAHI_GLIB_REQUIRED=0.6
