From 599f3e2020dc3164f31ca7012422f2b9b81d85bc Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Fri, 5 Jan 2018 13:30:32 -0500
Subject: [PATCH 01/12] Disable modules and SSL

---
 setup.py | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index e74a275edb..595be3ddd7 100644
--- a/setup.py
+++ b/setup.py
@@ -58,7 +58,17 @@
 TEST_EXTENSIONS = (sysconfig.get_config_var('TEST_MODULES') == 'yes')
 
 # This global variable is used to hold the list of modules to be disabled.
-DISABLED_MODULE_LIST = []
+pdm_env = "PYTHON_DISABLE_MODULES"
+if pdm_env in os.environ:
+    DISABLED_MODULE_LIST = os.environ[pdm_env].split()
+else:
+    DISABLED_MODULE_LIST = []
+
+pds_env = "PYTHON_DISABLE_SSL"
+if pds_env in os.environ:
+    disable_ssl = os.environ[pds_env]
+else:
+    disable_ssl = 0
 
 # --list-module-names option used by Tools/scripts/generate_module_names.py
 LIST_MODULE_NAMES = False
@@ -2435,6 +2445,7 @@ def detect_decimal(self):
                            depends=depends))
 
     def detect_openssl_hashlib(self):
+        global disable_ssl
         # Detect SSL support for the socket module (via _ssl)
         config_vars = sysconfig.get_config_vars()
 
@@ -2453,7 +2464,7 @@ def split_var(name, sep):
         openssl_libdirs = split_var('OPENSSL_LDFLAGS', '-L')
         openssl_libs = split_var('OPENSSL_LIBS', '-l')
         openssl_rpath = config_vars.get('OPENSSL_RPATH')
-        if not openssl_libs:
+        if not openssl_libs or disable_ssl:
             # libssl and libcrypto not found
             self.missing.extend(['_ssl', '_hashlib'])
             return None, None
-- 
2.35.1

