From f4ea33683d517ae9cc8d9e28f3c443393a0b2bf6 Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Wed, 4 Jul 2018 11:39:21 -0400
Subject: [PATCH 05/12] Disable nis

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index 252a27993b..7e72eafedc 100644
--- a/setup.py
+++ b/setup.py
@@ -1902,7 +1902,6 @@ def detect_modules(self):
         self.detect_dbm_gdbm()
         self.detect_sqlite()
         self.detect_platform_specific_exts()
-        self.detect_nis()
         self.detect_compress_exts()
         self.detect_expat_elementtree()
         self.detect_multibytecodecs()
-- 
2.35.1

