From dd2d7ff21ac218482e18b15a833cfe9256da979c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Tue, 14 Jul 2020 18:05:32 +0200
Subject: [PATCH 09/12] Skip RDS socket tests that hang frequently

---
 Lib/test/test_socket.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Lib/test/test_socket.py b/Lib/test/test_socket.py
index 5c15648b60..e054f7dcc5 100755
--- a/Lib/test/test_socket.py
+++ b/Lib/test/test_socket.py
@@ -2239,6 +2239,7 @@ def testSocketBufferSize(self):
 
 
 @unittest.skipUnless(HAVE_SOCKET_RDS, 'RDS sockets required for this test.')
+@unittest.skip('Hangs frequently')
 class RDSTest(ThreadedRDSSocketTest):
 
     def __init__(self, methodName='runTest'):
-- 
2.35.1

