From 6333e49ba67a7991b9cd8351026ad126af770cd2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Wed, 3 Feb 2021 10:14:49 +0100
Subject: [PATCH 10/12] ssl: Hard-disable SSLv3 to avoid automagic deps

Bug: https://bugs.gentoo.org/767886
---
 Modules/_ssl.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Modules/_ssl.c b/Modules/_ssl.c
index af2520432a..3fc3973b54 100644
--- a/Modules/_ssl.c
+++ b/Modules/_ssl.c
@@ -67,6 +67,10 @@
 #  error "OPENSSL_THREADS is not defined, Python requires thread-safe OpenSSL"
 #endif
 
+#ifndef OPENSSL_NO_SSL3
+#  define OPENSSL_NO_SSL3 1
+#endif
+
 
 
 struct py_ssl_error_code {
-- 
2.35.1

