diff -urN gnome-session-3.16.0/config.h.in gnome-session-3.16.0-patched/config.h.in
--- gnome-session-3.16.0/config.h.in	2015-03-23 13:27:54.000000000 -0400
+++ gnome-session-3.16.0-patched/config.h.in	2015-06-01 14:32:17.845912356 -0400
@@ -1,5 +1,8 @@
 /* config.h.in.  Generated from configure.ac by autoheader.  */
 
+/* Define if we should build deprecated functionalty */
+#undef ENABLE_DEPRECATED
+
 /* Define if IPV6 is supported */
 #undef ENABLE_IPV6
 
@@ -42,9 +45,6 @@
 /* Define to 1 if you have the <memory.h> header file. */
 #undef HAVE_MEMORY_H
 
-/* Define if we have an older upower */
-#undef HAVE_OLD_UPOWER
-
 /* Define to 1 if you have the <stdint.h> header file. */
 #undef HAVE_STDINT_H
 
diff -urN gnome-session-3.16.0/configure gnome-session-3.16.0-patched/configure
--- gnome-session-3.16.0/configure	2015-03-23 13:27:53.000000000 -0400
+++ gnome-session-3.16.0-patched/configure	2015-06-01 14:32:17.848912339 -0400
@@ -731,8 +731,8 @@
 X11_CFLAGS
 SESSION_PROPERTIES_LIBS
 SESSION_PROPERTIES_CFLAGS
-HAVE_OLD_UPOWER_FALSE
-HAVE_OLD_UPOWER_TRUE
+ENABLE_DEPRECATED_FALSE
+ENABLE_DEPRECATED_TRUE
 UPOWER_LIBS
 UPOWER_CFLAGS
 GNOME_SESSION_LIBS
@@ -886,6 +886,7 @@
 enable_iso_c
 enable_deprecation_flags
 enable_session_selector
+enable_deprecated
 enable_gconf
 enable_systemd
 enable_consolekit
@@ -1585,6 +1586,7 @@
                           use *_DISABLE_DEPRECATED flags [default=no]
   --enable-session-selector
                           enable building a custom session selector dialog
+  --enable-deprecated     build deprecated functionality
   --enable-gconf          Support gconf-based autostart
   --enable-systemd        Use systemd
   --enable-consolekit     Use consolekit
@@ -12722,90 +12724,25 @@
 
 fi
 
-
-pkg_failed=no
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for UPOWER" >&5
-$as_echo_n "checking for UPOWER... " >&6; }
-
-if test -n "$UPOWER_CFLAGS"; then
-    pkg_cv_UPOWER_CFLAGS="$UPOWER_CFLAGS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"upower-glib < 0.99.0\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "upower-glib < 0.99.0") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-  pkg_cv_UPOWER_CFLAGS=`$PKG_CONFIG --cflags "upower-glib < 0.99.0" 2>/dev/null`
-		      test "x$?" != "x0" && pkg_failed=yes
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-if test -n "$UPOWER_LIBS"; then
-    pkg_cv_UPOWER_LIBS="$UPOWER_LIBS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"upower-glib < 0.99.0\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "upower-glib < 0.99.0") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-  pkg_cv_UPOWER_LIBS=`$PKG_CONFIG --libs "upower-glib < 0.99.0" 2>/dev/null`
-		      test "x$?" != "x0" && pkg_failed=yes
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-
-
-
-if test $pkg_failed = yes; then
-   	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-
-if $PKG_CONFIG --atleast-pkgconfig-version 0.20; then
-        _pkg_short_errors_supported=yes
+# Build deprecated functionality
+# Check whether --enable-deprecated was given.
+if test "${enable_deprecated+set}" = set; then :
+  enableval=$enable_deprecated; enable_deprecated=$enableval
 else
-        _pkg_short_errors_supported=no
+  enable_deprecated=no
 fi
-        if test $_pkg_short_errors_supported = yes; then
-	        UPOWER_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "upower-glib < 0.99.0" 2>&1`
-        else
-	        UPOWER_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "upower-glib < 0.99.0" 2>&1`
-        fi
-	# Put the nasty error message in config.log where it belongs
-	echo "$UPOWER_PKG_ERRORS" >&5
-
-	have_old_upower=no
-elif test $pkg_failed = untried; then
-     	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-	have_old_upower=no
-else
-	UPOWER_CFLAGS=$pkg_cv_UPOWER_CFLAGS
-	UPOWER_LIBS=$pkg_cv_UPOWER_LIBS
-        { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-	have_old_upower=yes
-fi
-if test x$have_old_upower = xyes; then :
-
 
-$as_echo "#define HAVE_OLD_UPOWER 1" >>confdefs.h
+if test "${enable_deprecated}" != no; then
 
+$as_echo "#define ENABLE_DEPRECATED 1" >>confdefs.h
 
 fi
- if test x$have_old_upower = xyes; then
-  HAVE_OLD_UPOWER_TRUE=
-  HAVE_OLD_UPOWER_FALSE='#'
+ if test x$enable_deprecated = xyes; then
+  ENABLE_DEPRECATED_TRUE=
+  ENABLE_DEPRECATED_FALSE='#'
 else
-  HAVE_OLD_UPOWER_TRUE='#'
-  HAVE_OLD_UPOWER_FALSE=
+  ENABLE_DEPRECATED_TRUE='#'
+  ENABLE_DEPRECATED_FALSE=
 fi
 
 
@@ -16327,8 +16264,8 @@
   as_fn_error $? "conditional \"BUILD_SESSION_SELECTOR\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
 fi
-if test -z "${HAVE_OLD_UPOWER_TRUE}" && test -z "${HAVE_OLD_UPOWER_FALSE}"; then
-  as_fn_error $? "conditional \"HAVE_OLD_UPOWER\" was never defined.
+if test -z "${ENABLE_DEPRECATED_TRUE}" && test -z "${ENABLE_DEPRECATED_FALSE}"; then
+  as_fn_error $? "conditional \"ENABLE_DEPRECATED\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
 fi
 if test -z "${HAVE_SYSTEMD_TRUE}" && test -z "${HAVE_SYSTEMD_FALSE}"; then
@@ -18653,7 +18590,7 @@
         XRender support:          ${have_xrender}
         XSync support:            ${have_xsync}
         XTest support:            ${have_xtest}
-	Legacy UPower backend:    ${have_old_upower}
+        Enable deprecated stuff:  ${enable_deprecated}
         Build Docbook:            ${enable_docbook_docs}
         Build manpages:           ${enable_man}
 
diff -urN gnome-session-3.16.0/configure.ac gnome-session-3.16.0-patched/configure.ac
--- gnome-session-3.16.0/configure.ac	2015-03-23 13:27:18.000000000 -0400
+++ gnome-session-3.16.0-patched/configure.ac	2015-06-01 14:32:17.848912339 -0400
@@ -67,13 +67,14 @@
         gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED
 )
 
-dnl We can only support old upower
-dnl https://bugzilla.gnome.org/show_bug.cgi?id=710383
-PKG_CHECK_MODULES(UPOWER, upower-glib < 0.99.0, have_old_upower=yes, have_old_upower=no)
-AS_IF([test x$have_old_upower = xyes], [
-  AC_DEFINE([HAVE_OLD_UPOWER], [1], [Define if we have an older upower])
-])
-AM_CONDITIONAL(HAVE_OLD_UPOWER, test x$have_old_upower = xyes)
+# Build deprecated functionality
+AC_ARG_ENABLE(deprecated, [  --enable-deprecated     build deprecated functionality],
+             enable_deprecated=$enableval,
+             enable_deprecated=no)
+if test "${enable_deprecated}" != no; then
+  AC_DEFINE(ENABLE_DEPRECATED, 1, [Define if we should build deprecated functionalty])
+fi
+AM_CONDITIONAL(ENABLE_DEPRECATED, test x$enable_deprecated = xyes)
 
 PKG_CHECK_MODULES(SESSION_PROPERTIES,
         glib-2.0 >= $GLIB_REQUIRED
@@ -413,7 +414,7 @@
         XRender support:          ${have_xrender}
         XSync support:            ${have_xsync}
         XTest support:            ${have_xtest}
-	Legacy UPower backend:    ${have_old_upower}
+        Enable deprecated stuff:  ${enable_deprecated}
         Build Docbook:            ${enable_docbook_docs}
         Build manpages:           ${enable_man}
 
diff -urN gnome-session-3.16.0/gnome-session/Makefile.am gnome-session-3.16.0-patched/gnome-session/Makefile.am
--- gnome-session-3.16.0/gnome-session/Makefile.am	2014-12-12 10:15:16.000000000 -0500
+++ gnome-session-3.16.0-patched/gnome-session/Makefile.am	2015-06-01 14:32:17.866912233 -0400
@@ -11,7 +11,7 @@
 AM_CPPFLAGS =					\
 	$(GTK3_CFLAGS)				\
 	$(GNOME_SESSION_CFLAGS)			\
-	$(UPOWER_CFLAGS)			\
+	-I/usr/include/libupower-glib			\
 	$(DISABLE_DEPRECATED_CFLAGS)
 
 AM_CFLAGS = $(WARN_CFLAGS)
@@ -93,7 +93,7 @@
 	$(XTEST_LIBS)				\
 	$(XEXT_LIBS)				\
 	$(GNOME_SESSION_LIBS)			\
-	$(UPOWER_LIBS)				\
+	-lupower-glib				\
 	$(GCONF_LIBS)				\
 	$(CONSOLEKIT_LIBS)			\
 	$(SYSTEMD_LIBS)				\
diff -urN gnome-session-3.16.0/gnome-session/gsm-consolekit.c gnome-session-3.16.0-patched/gnome-session/gsm-consolekit.c
--- gnome-session-3.16.0/gnome-session/gsm-consolekit.c	2014-12-12 10:23:26.000000000 -0500
+++ gnome-session-3.16.0-patched/gnome-session/gsm-consolekit.c	2015-06-01 14:36:07.602558691 -0400
@@ -29,10 +29,9 @@
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
-#ifdef HAVE_OLD_UPOWER
-#define UPOWER_ENABLE_DEPRECATED 1
-#include <upower.h>
-#endif
+#ifdef ENABLE_DEPRECATED
+#include <libupower-glib/upower.h>
+#endif /* ENABLE_DEPRECATED */
 
 #include "gsm-system.h"
 #include "gsm-consolekit.h"
@@ -56,9 +55,9 @@
         DBusGProxy      *bus_proxy;
         DBusGProxy      *ck_proxy;
         DBusGProxy      *session_proxy;
-#ifdef HAVE_OLD_UPOWER
+#ifdef ENABLE_DEPRECATED
         UpClient        *up_client;
-#endif
+#endif /* ENABLE_DEPRECATED */
 
         gboolean         is_active;
         gboolean         restarting;
@@ -308,10 +307,10 @@
                                              G_CALLBACK (on_active_changed), manager, NULL);
         }
 
-#ifdef HAVE_OLD_UPOWER
+#ifdef ENABLE_DEPRECATED
         g_clear_object (&manager->priv->up_client);
         manager->priv->up_client = up_client_new ();
-#endif
+#endif /* ENABLE_DEPRECATED */
 
         is_connected = TRUE;
 
@@ -385,9 +384,9 @@
         g_clear_object (&manager->priv->ck_proxy);
         g_clear_object (&manager->priv->session_proxy);
         g_clear_object (&manager->priv->session_id);
-#ifdef HAVE_OLD_UPOWER
+#ifdef ENABLE_DEPRECATED
         g_clear_object (&manager->priv->up_client);
-#endif
+#endif /* ENABLE_DEPRECATED */
 
         if (manager->priv->dbus_connection != NULL) {
                 DBusConnection *connection;
@@ -954,29 +953,29 @@
 static gboolean
 gsm_consolekit_can_suspend (GsmSystem *system)
 {
-#ifdef HAVE_OLD_UPOWER
+#ifdef ENABLE_DEPRECATED
         GsmConsolekit *consolekit = GSM_CONSOLEKIT (system);
         return up_client_get_can_suspend (consolekit->priv->up_client);
 #else
         return FALSE;
-#endif
+#endif /* ENABLE_DEPRECATED */
 }
 
 static gboolean
 gsm_consolekit_can_hibernate (GsmSystem *system)
 {
-#ifdef HAVE_OLD_UPOWER
+#ifdef ENABLE_DEPRECATED
         GsmConsolekit *consolekit = GSM_CONSOLEKIT (system);
         return up_client_get_can_hibernate (consolekit->priv->up_client);
 #else
         return FALSE;
-#endif
+#endif /* ENABLE_DEPRECATED */
 }
 
 static void
 gsm_consolekit_suspend (GsmSystem *system)
 {
-#ifdef HAVE_OLD_UPOWER
+#ifdef ENABLE_DEPRECATED
         GsmConsolekit *consolekit = GSM_CONSOLEKIT (system);
         GError *error = NULL;
         gboolean ret;
@@ -986,13 +985,13 @@
                 g_warning ("Unexpected suspend failure: %s", error->message);
                 g_error_free (error);
         }
-#endif
+#endif /* ENABLE_DEPRECATED */
 }
 
 static void
 gsm_consolekit_hibernate (GsmSystem *system)
 {
-#ifdef HAVE_OLD_UPOWER
+#ifdef ENABLE_DEPRECATED
         GsmConsolekit *consolekit = GSM_CONSOLEKIT (system);
         GError *error = NULL;
         gboolean ret;
@@ -1002,7 +1001,7 @@
                 g_warning ("Unexpected hibernate failure: %s", error->message);
                 g_error_free (error);
         }
-#endif
+#endif /* ENABLE_DEPRECATED */
 }
 
 static void
