diff --git a/glib/gtester-report b/glib/gtester-report
index 7d3139f58..1891de67d 100755
--- a/glib/gtester-report
+++ b/glib/gtester-report
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python
 # GLib Testing Framework Utility			-*- Mode: python; -*-
 # Copyright (C) 2007 Imendio AB
 # Authors: Tim Janik
diff --git a/gobject/glib-genmarshal.in b/gobject/glib-genmarshal.in
index 2560f9fb2..aeb1c8695 100755
--- a/gobject/glib-genmarshal.in
+++ b/gobject/glib-genmarshal.in
@@ -1,4 +1,4 @@
-#!/usr/bin/env @PYTHON@
+#!/usr/bin/python-exec2c
 
 # pylint: disable=too-many-lines, missing-docstring, invalid-name
 
diff --git a/gobject/glib-mkenums.in b/gobject/glib-mkenums.in
index d551cdcf4..3949c7404 100755
--- a/gobject/glib-mkenums.in
+++ b/gobject/glib-mkenums.in
@@ -1,4 +1,4 @@
-#!/usr/bin/env @PYTHON@
+#!/usr/bin/python-exec2c
 
 # If the code below looks horrible and unpythonic, do not panic.
 #
