--- a/jpeg-8b/jmorecfg.h
+++ b/jpeg-8b/jmorecfg.h
@@ -268,14 +268,14 @@
 /* Capability options common to encoder and decoder: */
 
 #define DCT_ISLOW_SUPPORTED	/* slow but accurate integer algorithm */
-#define DCT_IFAST_SUPPORTED	/* faster, less accurate integer method */
-#define DCT_FLOAT_SUPPORTED	/* floating-point: accurate, fast on fast HW */
+#undef  DCT_IFAST_SUPPORTED	/* faster, less accurate integer method */
+#undef  DCT_FLOAT_SUPPORTED	/* floating-point: accurate, fast on fast HW */
 
 /* Encoder capability options: */
 
-#define C_ARITH_CODING_SUPPORTED    /* Arithmetic coding back end? */
-#define C_MULTISCAN_FILES_SUPPORTED /* Multiple-scan JPEG files? */
-#define C_PROGRESSIVE_SUPPORTED	    /* Progressive JPEG? (Requires MULTISCAN)*/
+#undef  C_ARITH_CODING_SUPPORTED    /* Arithmetic coding back end? */
+#undef  C_MULTISCAN_FILES_SUPPORTED /* Multiple-scan JPEG files? */
+#undef  C_PROGRESSIVE_SUPPORTED	    /* Progressive JPEG? (Requires MULTISCAN)*/
 #define DCT_SCALING_SUPPORTED	    /* Input rescaling via DCT? (Requires DCT_ISLOW)*/
 #define ENTROPY_OPT_SUPPORTED	    /* Optimization of entropy coding parms? */
 /* Note: if you selected 12-bit data precision, it is dangerous to turn off
@@ -286,7 +286,7 @@
  * The exact same statements apply for progressive JPEG: the default tables
  * don't work for progressive mode.  (This may get fixed, however.)
  */
-#define INPUT_SMOOTHING_SUPPORTED   /* Input image smoothing option? */
+#undef  INPUT_SMOOTHING_SUPPORTED   /* Input image smoothing option? */
 
 /* Decoder capability options: */
 
@@ -295,11 +295,11 @@
 #define D_PROGRESSIVE_SUPPORTED	    /* Progressive JPEG? (Requires MULTISCAN)*/
 #define IDCT_SCALING_SUPPORTED	    /* Output rescaling via IDCT? */
 #define SAVE_MARKERS_SUPPORTED	    /* jpeg_save_markers() needed? */
-#define BLOCK_SMOOTHING_SUPPORTED   /* Block smoothing? (Progressive only) */
+#undef  BLOCK_SMOOTHING_SUPPORTED   /* Block smoothing? (Progressive only) */
 #undef  UPSAMPLE_SCALING_SUPPORTED  /* Output rescaling at upsample stage? */
 #define UPSAMPLE_MERGING_SUPPORTED  /* Fast path for sloppy upsampling? */
-#define QUANT_1PASS_SUPPORTED	    /* 1-pass color quantization? */
-#define QUANT_2PASS_SUPPORTED	    /* 2-pass color quantization? */
+#undef  QUANT_1PASS_SUPPORTED	    /* 1-pass color quantization? */
+#undef  QUANT_2PASS_SUPPORTED	    /* 2-pass color quantization? */
 
 /* more capability options later, no doubt */
 
--- a/libpng-1.6.37/pngusr.dfa
+++ b/libpng-1.6.37/pngusr.dfa
@@ -1,14 +1,52 @@
 # pngusr.dfa
 #
-# Build time configuration of libpng
+# Build time configuration of libpng for Squeezebox Server
 #
-# Enter build configuration options in this file
-#
-# Security settings: by default these limits are unset, you can change them
-# here by entering the appropriate values as #defines preceded by '@' (to cause,
-# them to be passed through to the build of pnglibconf.h), for example:
-#
-# @# define PNG_USER_WIDTH_MAX 65535
-# @# define PNG_USER_HEIGHT_MAX 65535
-# @# define PNG_USER_CHUNK_CACHE_MAX 256
-# @# define PNG_USER_CHUNK_MALLOC_MAX 640000
+# Tested with libpng-1.6.36 and Image::Scale v0.14
+option MNG_FEATURES off
+option FLOATING_ARITHMETIC off
+option FLOATING_POINT off
+option FIXED_POINT on
+option CONSOLE_IO off
+option READ_TEXT off
+option READ_hIST off
+option READ_iCCP off
+option READ_oFFs off
+option READ_pCAL off
+option READ_sCAL off
+option READ_pHYs off
+option READ_sBIT off
+option READ_sPLT off
+option READ_sRGB off
+option READ_tIME off
+option UNKNOWN_CHUNKS off
+
+option WRITE_cHRM off
+option WRITE_bKGD off
+option WRITE_gAMA off
+option WRITE_hIST off
+option WRITE_iCCP off
+option WRITE_oFFs off
+option WRITE_pCAL off
+option WRITE_sCAL off
+option WRITE_pHYs off
+option WRITE_sPLT off
+option WRITE_sRGB off
+option WRITE_tIME off
+option INFO_IMAGE off
+
+# These are all the options that were included as part of the libpng-1.4.3 conf
+# authored by agrundman. They are commented out because they prevent compilation
+# or prevent compilation of Image::Scale. Enable them at your own peril.
+
+# Disabling any of the following prevents libpng from compiling
+#option READ_cHRM off
+#option WRITE_TEXT off
+#option WRITE_sBIT off
+
+# Disabling any of the following causes libpng to fail one or more tests
+#option READ_gAMA off
+#option WRITE_tRNS off
+
+# Disabling any of the following causes Image::Scale to fail compilation
+#option STDIO off
