--- a/bin/joomla 2017-08-01 15:37:31.281807945 +0000
+++ b/bin/joomla      2017-08-01 15:40:36.257240446 +0000
@@ -11,7 +11,8 @@
     __DIR__ . '/../../../../vendor/autoload.php',
     __DIR__ . '/../../../autoload.php',
     __DIR__ . '/../vendor/autoload.php',
-    __DIR__ . '/vendor/autoload.php'
+    __DIR__ . '/vendor/autoload.php',
+       __DIR__ . '/../share/php/Joomlatools/Console/autoload.php'
 );
 
 foreach ($autoloadPaths as $autoload) {

