From c4ce0e3eadad4b16983c7f706be91dea36bc2092 Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Fri, 5 Jan 2018 13:30:32 -0500
Subject: [PATCH 02/14] Disable modules and SSL

---
 setup.py | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 0bec170d3f..309f95806a 100644
--- a/setup.py
+++ b/setup.py
@@ -44,7 +44,17 @@ from distutils.spawn import find_executable
 TEST_EXTENSIONS = True
 
 # This global variable is used to hold the list of modules to be disabled.
-DISABLED_MODULE_LIST = []
+pdm_env = "PYTHON_DISABLE_MODULES"
+if pdm_env in os.environ:
+    DISABLED_MODULE_LIST = os.environ[pdm_env].split()
+else:
+    DISABLED_MODULE_LIST = []
+
+pds_env = "PYTHON_DISABLE_SSL"
+if pds_env in os.environ:
+    disable_ssl = os.environ[pds_env]
+else:
+    disable_ssl = 0
 
 
 def get_platform():
@@ -2316,6 +2326,7 @@ class PyBuildExt(build_ext):
                            depends=depends))
 
     def detect_openssl_hashlib(self):
+        global disable_ssl
         # Detect SSL support for the socket module (via _ssl)
         config_vars = sysconfig.get_config_vars()
 
@@ -2333,7 +2344,7 @@ class PyBuildExt(build_ext):
         openssl_includes = split_var('OPENSSL_INCLUDES', '-I')
         openssl_libdirs = split_var('OPENSSL_LDFLAGS', '-L')
         openssl_libs = split_var('OPENSSL_LIBS', '-l')
-        if not openssl_libs:
+        if not openssl_libs or disable_ssl:
             # libssl and libcrypto not found
             self.missing.extend(['_ssl', '_hashlib'])
             return None, None
-- 
2.35.1

