From 9ec8f9b71648a8c0b82544c1e468afbcff6a5963 Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Wed, 4 Jul 2018 11:39:21 -0400
Subject: [PATCH 06/14] Disable nis

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index e0fee3c60d..552a6ade14 100644
--- a/setup.py
+++ b/setup.py
@@ -1810,7 +1810,6 @@ class PyBuildExt(build_ext):
         self.detect_dbm_gdbm()
         self.detect_sqlite()
         self.detect_platform_specific_exts()
-        self.detect_nis()
         self.detect_compress_exts()
         self.detect_expat_elementtree()
         self.detect_multibytecodecs()
-- 
2.35.1

