diff --git a/src/load_pat.cpp b/src/load_pat.cpp
index adcccc3..1d055e0 100644
--- a/src/load_pat.cpp
+++ b/src/load_pat.cpp
@@ -52,8 +52,8 @@
 #define PATHFORPAT	"C:\\TIMIDITY\\INSTRUMENTS"
 #else
 #define DIRDELIM		'/'
-#define TIMIDITYCFG	"/usr/local/share/timidity/timidity.cfg"
-#define PATHFORPAT	"/usr/local/share/timidity/instruments"
+#define TIMIDITYCFG	"/usr/share/timidity/timidity.cfg"
+#define PATHFORPAT	"/usr/share/timidity/instruments"
 #endif
 
 #define PAT_ENV_PATH2CFG			"MMPAT_PATH_TO_CFG"
