diff -uNr nvi-1.81.6.orig/dist/configure.in nvi-1.81.6/dist/configure.in
--- nvi-1.81.6.orig/dist/configure.in	2007-11-18 17:41:42.000000000 +0100
+++ nvi-1.81.6/dist/configure.in	2022-02-15 22:23:36.504788790 +0100
@@ -790,7 +790,7 @@
 dnl Check for which version of openpty to use, System V or Berkeley.
 AC_MSG_CHECKING(for System V pty calls)
 AC_CACHE_VAL(vi_cv_sys5_pty, [dnl
-AC_TRY_LINK(, [grantpt(0);],
+AC_TRY_LINK([#include <sys/stropts.h>], [grantpt(0);],
 	[vi_cv_sys5_pty=yes], [vi_cv_sys5_pty=no])])
 if test "$vi_cv_sys5_pty" = yes; then
 	AC_DEFINE(HAVE_SYS5_PTY)
