
--- a/layman/argsparser.py	2019-03-06 15:08:03.000000000 +0100
+++ b/layman/argsparser.py	2019-03-09 15:31:58.296182966 +0100
@@ -364,6 +364,8 @@
                 protocol_filter = self.options[key]
             if self.config.has_option('MAIN', 'protocol_filter'):
                 protocol_filter = self.config.get('MAIN', 'protocol_filter')
+                if  isinstance(protocol_filter, str) and protocol_filter=='[]' :
+                  protocol_filter= []
             if protocol_filter:
                 if not isinstance(protocol_filter, list):
                     protocol_filter = [e.strip() for e in protocol_filter.split(',')]
